---
layout: news
tags: [HADDOCK, EVFold, Co-Evolution, Docking, Integrative Modeling, Bioinformatics, Harvard University, Utrecht University]
modified: 2014-08-08T20:53:07.573882-04:00
comments: false
excerpt: 'HADDOCKing with co-evolution restraints: collaboration with Harvard University @eLife'
image:
  feature: posts/banner_once-upon-a-time-two-proteins.jpg
---

Darwin is well known for his natural selection theory, the trigger for the
acceptance of evolution as a major driving force in the biological world.
Alongside natural selection, Darwin proposed the theory of co-evolution
to explain concerted adaptations he observed between species. At the
molecular level, co-evolution is perceived in compensatory mutations in
protein sequences, often necessary to maintain the physical interactions that
define protein structure and interactions. In a recent publication in eLife,
researchers from Utrecht University, in collaboration with groups at Harvard
University and the Memorial Sloan Kettering Cancer Center, showed that
it is possible to mathematically exploit this co-evolution phenomenon and
use it to infer, only from sequence information, the pairs of amino acids that
define the interaction between two proteins. Combining this information with
data-driven computational docking approaches such as [HADDOCK](http:/
/haddock.science.uu.nl), developed by Prof. Alexandre Bonvin at Utrecht
University, allows accurate modeling of three-dimensional structures of protein
interactions. This work is a breakthrough in protein interaction modeling,
since current methodologies rely on experimental data, often hard to obtain, to
produce accurate models. As such, this possibility of exploiting the evolutionary
record of proteins to model their interactions paves the way for a low-cost but
high-throughput alternative with broad applications, namely drug design.

* T.A. Hopf#, C.P.I. Sch&auml;rfe#, J.P.G.L.M. Rodrigues#, A.G. Green, O. Kohlbacher, C. Sander\*, **A.M.J.J. Bonvin\*** and D.S. Marks\*.
[Sequence co-evolution gives 3D contacts and structures of protein complexes.](http://elifesciences.org/content/early/2014/09/25/eLife.03430)
_eLife._ 10.7554/eLife.03430, Advanced Online Publication (2014)
