---
layout: page
tags: [Jekyll, HADDOCK, Bonvin, Docking, Simulation, Molecular Dynamics, Structural Biology, Computational Biology, Modelling, Protein Structure]
modified: 2014-08-08T20:53:07.573882-04:00
comments: false
title: HADDOCK-related publications
image:
  feature: pages/banner_software.jpg
---


* * *

This page lists haddock development articles.  

The original JACS paper, together with an introduction to the study of biomolecular complexes and an application to the UbcH5-CNOT4 complex can be found online in Cyril Dominguez's Ph.D. thesis:

*   ["NMR-based docking of protein-protein complexes: the human UbcH5b-CNOT4 ubiquitination complex"](https://www.library.uu.nl/digiarchief/dip/diss/2004-0616-081333/inhoud.htm),<br>
      Cyril Dominguez, Ph.D. Thesis, June 2004, ISBN 90-393-3721-7.

**_When using HADDOCK cite:_**

*   Cyril Dominguez, Rolf Boelens and **Alexandre M.J.J. Bonvin** (2003). [HADDOCK: a protein-protein docking approach based on biochemical and/or biophysical information.](https://doi.org/doi:10.1021/ja026939x)  
    _J. Am. Chem. Soc._ **125**, 1731-1737.

*   S.J. de Vries, A.D.J. van Dijk, M. Krzeminski, M. van Dijk, A. Thureau, V. Hsu, T. Wassenaar and **A.M.J.J. Bonvin**  
    "[HADDOCK versus HADDOCK: New features and performance of HADDOCK2.0 on the CAPRI targets.](https://doi.org/doi:10.1002/prot.21723)"  
    _Proteins: Struc. Funct. & Bioinformatic_ **69**, 726-733 (2007).

**_When using the HADDOCK2.4 web service cite:_**

* R.V. Honorato, M.E. Trellet, B. Jiménez-García1, J.J. Schaarschmidt, M. Giulini, V. Reys,  P.I. Koukos, J.P.G.L.M. Rodrigues, E. Karaca, G.C.P. van Zundert, J. Roel-Touris, C.W. van Noort, Z. Jandová, A.S.J. Melquiond and **A.M.J.J. Bonvin**. [The HADDOCK2.4 web server: A leap forward in integrative modelling of biomolecular complexes](https://www.nature.com/articles/s41596-024-01011-0.epdf?sharing_token=UHDrW9bNh3BqijxD2u9Xd9RgN0jAjWel9jnR3ZoTv0O8Cyf_B_3QikVaNIBRHxp9xyFsQ7dSV3t-kBtpCaFZWPfnuUnAtvRG_vkef9o4oWuhrOLGbBXJVlaaA9ALOULn6NjxbiqC2VkmpD2ZR_r-o0sgRZoHVz10JqIYOeus_nM%3D). _Nature Prot._, Advanced Online Publication DOI: 10.1038/s41596-024-01011-0 (2024).


**_And, if used, cite depending on the type of data and setting used the following publications:_**  

The use of _residual dipolar couplings_ in HADDOCK is described in:  

*   A.D.J. van Dijk, D. Fushman and **A.M.J.J. Bonvin** (2005)  
    "[Various strategies of using residual dipolar couplings in NMR-driven protein docking: Application to Lys48-linked di-ubiquitin and validation against 15N-relaxation data.](https://doi.org/doi:10.1002/prot.20476)"  
    _Proteins: Struc. Funct. & Bioinformatics_, **60**, 367-381.

The use of _diffusion anisotropy data_ in HADDOCK is described in:  

*   A.D.J. van Dijk, R. Kaptein, R. Boelens and **A.M.J.J. Bonvin**
    "[Combining NMR relaxation with chemical shift perturbation data to drive protein-protein docking](https://doi.org/doi:10.1007/s10858-006-0024-8)"  
    _J. Biomol. NMR_, **34**, 237-244 (2006).

The use of _pseudo contact shift restraints_ in HADDOCK is described in:  

*   C. Schmitz and **A.M.J.J. Bonvin**  
    "[Protein-Protein HADDocking using exclusively Pseudocontact Shifts.](https://doi.org/doi:10.1007/s10858-011-9514-4)"  
    _J. Biomol. NMR_, **50**, 263-266 (2011).

The use of _SAXS data to score HADDOCK models_ is described in:  

*   E. Karaca and **A.M.J.J. Bonvin**.   
    "[On the usefulness of Ion Mobility Mass Spectrometry and SAXS data in scoring docking decoys.](https://doi.org/doi:10.1107/S0907444913007063)"  
    _Acta Cryst. D._, **D69**, 683-694 (2013).

The _quantitative use of chemical shifts to score HADDOCK models_ is described in:  

*   D. Stratman, R. Boelens and **A.M.J.J. Bonvin**.   
    "[Quantitative use of chemical shifts for the modelling of protein complexes.](https://doi.org/doi:10.1002/prot.23090)"  
    _Proteins: Struc. Funct. & Bioinformatic_, **79**, 2662-2670 (2011).

_Protein-peptide docking_ with HADDOCK is described in:  

*   M. Trellet, A.S.J. Melquiond and **A.M.J.J. Bonvin**. .  
    "[A Unified Conformational Selection and Induced Fit Approach to Protein-Peptide Docking.](https://dx.plos.org/10.1371/journal.pone.0058769)  
    _PLoS ONE_, **8(3)** e58769 (2013).

_Solvated protein-protein docking_ using HADDOCK is described in:  

*   A.D.J. van Dijk and **A.M.J.J. Bonvin**
    "[Solvated docking: introducing water i nto the modelling of biomolecular complexes.](https://doi.org/doi:10.1093/bioinformatics/btl395)"  
    _Bioinformatics_, **22** 2340-2347 (2006).

_Solvated protein-protein docking with Kyte-Doolittle propensities_ using HADDOCK is described in:  

*   P.L. Kastritis, K.M. Visscher, A.D.J. van Dijk and **A.M.J.J. Bonvin**.   
    "[Solvated docking using Kyte-Doolittle-based water propensities.](https://doi.org/doi:10.1002/prot.24210)"  
    _Proteins: Struc. Funct. & Bioinformatic._, **81**, 510-518 (2013).

_Solvated protein-DNA docking_ using HADDOCK is described in:  

*   M. van Dijk, K. Visscher, P.L. Kastritis and **A.M.J.J. Bonvin**. .  
    "[Solvated protein-DNA docking using HADDOCK.](https://doi.org/doi:10.1007/s10858-013-9734-x)"  
    _J. Biomol. NMR_, **56**, 51-63 (2013).

_Flexible protein-DNA docking_ using HADDOCK is described in:

*   M. van Dijk, A.D.J. van Dijk, V. Hsu, R. Boelens and **A.M.J.J. Bonvin**
    "[Information-driven Protein-DNA Docking using HADDOCK: it is a matter of flexibility.](https://doi.org/doi:10.1093/nar/gkl412)"  
    _Nucl. Acids Res._, **34** 3317-3325 (2006).  

*   M. van Dijk and **A.M.J.J. Bonvin**.   
    "[Pushing the limits of what is achievable in protein-DNA docking. Benchmarking HADDOCK's performance.](https://doi.org/doi:10.1093/nar/gkq222)"  
    _Nucl. Acid Res._, **38**, 5634-5647 (2010).

_(Flexible) multibody docking_ using HADDOCK is described in:  

*   E. Karaca and **A.M.J.J. Bonvin**.   
    "[A multi-domain flexible docking approach to deal with large conformational changes in the modeling of biomolecular complexes.](https://doi.org/doi:10.1016/j.str.2011.01.014)"  
    _Structure_, **19** 555-565(2011).  

*   E. Karaca, A.S.J. Melquiond, S.J. de Vries, P.L. Kastritis and **A.M.J.J. Bonvin**.   
    "[Building macromolecular assemblies by information-driven docking: Introducing the HADDOCK multi-body docking server.](https://doi.org/doi:10.1074/mcp.M000051-MCP201)"  
    _Mol. Cell. Proteomics_, **9**, 1784-1794 (2010).

_Our results in [CAPRI](https://capri.ebi.ac.uk)_, the blind docking experiment, are described in (Note that entire issues of Protein are dedicated to CAPRI):  

*   J.P.G.L.M. Rodrigues, A.S.J. Melquiond, E. Karaca, M. Trellet, M. Van Dijk, G.C.P. Van Zundert, C. Schmitz, S.J. de Vries, A. Bordogna, L. Bonati, P.L. Kastritis and **A.M.J.J. Bonvin**.   
    "[Defining the limits of homology modelling in information-driven protein docking](https://doi.org/doi:10.1002/prot.24382)"  
    _Proteins: Struc. Funct. & Bioinformatics_, **81**, 2119-2128 (2013).  

*   S.J. de Vries, A.S.J. Melquiond, P.L. Kastritis, E. Karaca, A. Bordogna, M. van Dijk, J.P.G.L.M. Rodrigues and **A.M.J.J. Bonvin**.   
    "[Strengths and weaknesses of data-driven docking in CAPRI](https://doi.org/doi:10.1002/prot.22814)"  
    _Proteins: Struc. Funct. & Bioinformatic_, **78**, 3242-3249 (2010).  

*   S.J. de Vries, A.D.J. van Dijk, M. Krzeminski, M. van Dijk, A. Thureau, V. Hsu, T. Wassenaar and **A.M.J.J. Bonvin**
    "[HADDOCK versus HADDOCK: New features and performance of HADDOCK2.0 on the CAPRI targets.](https://doi.org/doi:10.1002/prot.21723)"  
    _Proteins: Struc. Funct. & Bioinformatic_, advanced online publication (2007).  

*   A.D.J. van Dijk, S.J. de Vries, C. Dominguez, H. Chen, H.-X. Zhou and  **A.M.J.J. Bonvin**
    "[Data-driven docking: HADDOCKs adventures in CAPRI."](https://doi.org/doi:10.1002/prot.20563)  
    _Proteins: Struc. Funct. & Bioinformatics_, **60**, 232-238 (2005).  

Reviews on data-driven docking:

*   J.P.G.L.M Rodrigues and **A.M.J.J. Bonvin**.   
    "[Integrative computational modeling of protein interactions.](https://doi.org/doi:10.1111/febs.12771)"  
    _FEBS J._, **281**, 1988-2003 (2014).  

*   H. van Ingen and **A.M.J.J. Bonvin**.   
    "[Information-driven modeling of large macromolecular assemblies using NMR data.](https://doi.org/10.1016/j.jmr.2013.10.021)"  
    _J. Magn. Reson._ **241**, 103-114 (2014).  

*   E. Karaca and **A.M.J.J. Bonvin**.   
    "[Advances in integrated modelling of biomolecular complexes.](https://doi.org/doi:10.1016/j.ymeth.2012.12.004)"  
    _Methods_, **59**, 372-381 (2013).  

*   P.L. Kastritis and **A.M.J.J. Bonvin**.   
    "[Predicting and Dissecting High-order Molecular Complexity by Information-driven Biomolecular Docking](https://bookshop.cabi.org/?site=191&page=2633&pid=2443)"  
    In: _Antimicrobial Drug Discovery. Emerging Strategies_ Adv. Mol. Cell. Microbiol., Eds G. Tegos and E. Mylokanis, CABI, **22**, 232-246 (2012).  

*   A.S.J. Melquiond, E. Karaca, P.L. Kastritis and **A.M.J.J. Bonvin**.   
    "[Next challenges in protein-protein docking: From proteome to interactome and beyond.](https://onlinelibrary.wiley.com/doi/10.1002/wcms.91/full)"  
    _WIREs Computational Molecular Science_, **2**, 642-651 (2012).  

*   Aalt D.J. van Dijk, Rolf Boelens and **A.M.J.J. Bonvin** (2005). "[Data-driven docking for the study of biomolecular complexes.](https://doi.org/doi:10.1111/j.1742-4658.2004.04473.x)"  
    _FEBS Journal_, **272**, 293-312.

* * *
