---
layout: page
title: "Publications"
image:
  feature: pages/banner_publications-mini.jpg
---
## 2025

* **A.M.J.J. Bonvin** and M. Lensink. [Navigating the Pre- and Post-AlphaFold Divide: CAPRI 8th evaluation meeting, February 12-14, Grenoble, FR](https://doi.org/10.22541/au.175613473.32747661/v1), _Authorea_ Editorial to the Proteins 8th CAPRI special issue, 10.22541/au.175613473.32747661/v1 (2025).

* M. Sánchez-Marín,  M. Giuliniand **A.M.J.J. Bonvin**. [Combining AI structure prediction and integrative modelling for nanobody-antigen complexes](https://doi.org/10.1101/2025.07.01.662355). _BioRxiv_ 10.1101/2025.07.01.662355 (2025).

* A. Kryshtafovych, M. Milostan, M. Lensink, S. Velankar, **A.M.J.J. Bonvin**, J. Moult and K. Fidelis. [Updates to the CASP infrastructure in 2024](https://doi.org/10.1002/prot.70042). _Proteins: Struc. Funct. &amp; Bioinformatics_. Advanced Online Publication DOI:10.1002/prot.70042 (2025).

* M. Lensink, N. Raouraoua, G. Brysbaert, S. Velankar, S. Wodak and **A.M.J.J. Bonvin**. [Protein-protein interaction prediction in the pre- and
post-AlphaFold era: the 8th CAPRI evaluation](http://doi.org/10.1002/prot.70018). _Proteins: Struc. Funct. &amp; Bioinformatics_.  Advanced Online Publication DOI:10.1002/prot.70018 (2025)

* M. Giulini#, V. Reys#, J.M.C. Teixeira, B. Jiménez-García, R.V. Honorato, A. Kravchenko, X. Xu, R. Versini, A. Engel, S. Verhoeven and **A.M.J.J. Bonvin**. [HADDOCK3: A modular and versatile platform for integrative modelling of biomolecular complexes](https://doi.org/10.1021/acs.jcim.5c00969). _J. Chem. Inf. Mod._ *65*, 7315–7324 (2025). Preprint [here](https://doi.org/10.1101/2025.04.30.651432).

* X. Xu#, M. Giulini# and **A.M.J.J. Bonvin**. [Improved structural modelling of antibodies and their complexes with clustered diffusion ensembles](https://doi.org/10.1093/bioadv/vbaf161). _Bioinfo. Adv._ In Press, DOI:10.1093/bioadv/vbaf161 (2025). Preprint [here](https://doi.org/10.1101/2025.02.24.639865)

* L. Pulido-Cortés, H. Gielingh, V. Thijssen, M. Liu, R. Yoshisada, L.Romão Soares, S. Nizamuddin, F. Friedrich, H. Greschik, L. Peng, R. Vargas Honorato, M. Jung, **A.M.J.J. Bonvin**, M. Biniossek, R. Schüle, S. Jongkees, H. van Ingen, Hugo and H.Th.M. Timmers. [Molecular determinants for recognition of serotonylated chromatin](https://doi.org/10.1093/nar/gkaf612). _Nucl. Acids Res_. *53*, gkaf612 (2025).

* K. Devantier, T.L. Toft-Bertelsen, A. Prestel, V.M.S. Kjaer, C.a Sahin, M. Giulini, S. Louka, K. Spiess, A. Manandhar, K. Qvortrup, T. Ulven, B. Hjorth Bentzen, **A.M.J.J. Bonvin**, N. MacAulay, B.B. Kragelund and M.M. Rosenkilde. [The SH Protein of Mumps Virus is a Druggable Pentameric Viroporin](https://doi.org/10.1126/sciadv.ads3071). _Science Adv._ *11*, eads3071, 1-12 (2025). Preprint [here](https://doi.org/10.1101/2024.08.09.60700).

* A. Basciu, M. Athar, H. Kurt, C. Neville, G. Malloci, F. Muredda, A. Bosin, P. Ruggerone, **A.M.J.J. Bonvin** and A.V. Vargiu. [Predicting binding events in very flexible, allosteric, multi-domain proteins](https://pubs.acs.org/doi/10.1021/acs.jcim.4c01810). _J. Chem. Inf. Mod._ *65*, 2052–2065 (2025).

* X. Xu and **A.M.J.J. Bonvin** [Ranking protein-protein models with large language models and graph neural networks](https://doi.org/10.1007/978-1-0716-4623-6_4). In: KC, D.B. (eds) Large Language Models (LLMs) in Protein Bioinformatics. Methods in Molecular Biology, vol 2941, 71-83. Humana, New York, NY.71-83 (2025). Preprint [here](https://arxiv.org/abs/2407.16375)


## 2024

* V. Reys∗, Ma. Giulini∗, V. Cojocaru, A. Engel, X. Xu, J. Roel-Touris, C. Geng, F. Ambrosetti, B.
Jimenez-Garcia, Z. Jandova, P.I. Koukos, C. van Noort, J.M. . Teixeira, S.C. van Keulen, M. Reau, R.V. Honorato and **A.M.J.J. Bonvin**. [Integrative modeling in the age of machine learning: a summary of HADDOCK strategies in CAPRI rounds 47-55](http://doi.org/10.1002/prot.26789). _Proteins: Struc. Funct. &amp; Bioinformatics_.  Advanced Online Publication (2024).

* G. Bellinzona, D. Sassera and **A.M.J.J. Bonvin**. [Accelerating Protein-Protein Interaction screens with reduced AlphaFold-Multimer sampling](https://doi.org/10.1093/bioadv/vbae153) _Bioinformatics Advances_ *4*:vbae153 (2024).

* A. Ranaudo, M. Giulini, A. Pelissou Ayuso and **A.M.J.J. Bonvin**. [Modelling Protein-Glycan Interactions with HADDOCK](https://doi.org/10.1021/acs.jcim.4c01372). _J. Chem. Inf. Mod._ *64*, 7816–7825 (2024).

* M. Giulini, C. Schneider, D. Cutting, N. Desai, C. Deane and **A.M.J.J. Bonvin**. [Towards the accurate modelling of antibody-antigen complexes from sequence using machine learning and information-driven docking](https://doi.org/10.1093/bioinformatics/btae583). _Bioinformatics_ *40*:btae583, p. 1-11 (2024).[BioRxiv](https://www.biorxiv.org/content/10.1101/2023.11.17.567543v1)

* R.V. Honorato, M.E. Trellet, B. Jiménez-García1, J.J. Schaarschmidt, M. Giulini, V. Reys,  P.I. Koukos, J.P.G.L.M. Rodrigues, E. Karaca, G.C.P. van Zundert, J. Roel-Touris, C.W. van Noort, Z. Jandová, A.S.J. Melquiond and **A.M.J.J. Bonvin**. [The HADDOCK2.4 web server: A leap forward in integrative modelling of biomolecular complexes](https://www.nature.com/articles/s41596-024-01011-0.epdf?sharing_token=UHDrW9bNh3BqijxD2u9Xd9RgN0jAjWel9jnR3ZoTv0O8Cyf_B_3QikVaNIBRHxp9xyFsQ7dSV3t-kBtpCaFZWPfnuUnAtvRG_vkef9o4oWuhrOLGbBXJVlaaA9ALOULn6NjxbiqC2VkmpD2ZR_r-o0sgRZoHVz10JqIYOeus_nM%3D). _Nature Prot._, *19*, 3219–3241 (2024). Open access version [here](https://research-portal.uu.nl/ws/portalfiles/portal/245780710/s41596-024-01011-0_1_.pdf)

* B. Vallat, B.M. Webb, J.D. Westbrook, T. Goddard, C.A. Hanke, A. Graziadei, E. Peisach, A. Zalevsky, J. Sagendorf, H. Tangmunarunkit, S. Voinea, M. Sekharan, J. Yu, **A.M.J.J. Bonvin**, Fr, DiMaio, G. Hummer, J. Meiler, E. Tajkhorshid, T. Ferrin, C.L. Lawson, A. Leitner, J. Rappsilber, C.A.M. Seidel, C.M. Jeffries, S.K. Burley, J. Hoch, G.i Kurisu, K.e Morris, A. Patwardhan, S. Velankar, T. Schwede, J. Trewhella, C. Kesselman, H.M. Berman, A. Sali. [IHMCIF: An extension of PDBx/mmCIF data standard for integrative structure determination methods](https://doi.org/10.1016/j.jmb.2024.168546). _J. Mol. Biol._, *436*:168546 (2024).

* K.W. Collins, M.M. Copeland,, G. Brysbaert, S.J. Wodak, **A.M.J.J. Bonvin**, P,J, Kundrotas, I.A. Vakser and M.F. Lensink. [CAPRI-Q: The CAPRI resource evaluating the quality of predicted structures of protein complexes](https://doi.org/10.1016/j.jmb.2024.168540). _J. Mol. Biol._, *436*:168540 (2024).

* M. Giulini, R.V. Honorato, J.L. Rivera, **A.M.J.J. Bonvin**. [ARCTIC-3D: Automatic Retrieval and ClusTering of Interfaces in Complexes from 3D structural information](https://doi.org/10.1038/s42003-023-05718-w). _Comm. Biol._  *7*:49, p. 1-9  (2024).

* X. Xu, **A.M.J.J. Bonvin**. [DeepRank-GNN-esm: A Graph Neural Network for Scoring Protein-Protein Models using Protein Language Model](https://doi.org/10.1093/bioadv/vbad191). _Bioinfo. Adv._  *4*:vbad191  (2024).


## 2023

* **A.M.J.J. Bonvin**. [Empowering Global Collaboration in Structural Biology and Life Sciences](https://zenodo.org/record/8135315). DOI:10.5281/zenodo.8135315 (2023).

* M. Lensink, G. Brysbaert, N. Raouraoua, P. Bates, M. Giulini, R. Vargas Honorato, C. van Noort, J. Teixeira, **A.M.J.J. Bonvin**, R. Kong, H. Shi, X. Lu, S. Chang, J. Liu, Z. Guo, X. Chen, A. Morehead, R. Roy, T. Wu, N. Giri, F. Quadir, C. Chen, J. Cheng, C. Del Carpio, E. Ichiishi, L. Rodriguez-Lumbreras, J. Fernández-Recio, A. Harmalkar, L. Chu, S.Canner, R. Smanta, J. Gray, H. Li, P. Lin, J.a He, H. Tao, S. Huang, J. Roel, B. Jimenez-Garcia, C. Christoffer, A. Jain J, Y. Kagaya, H. Kannan, T. Nakamura, G. Terashi, J. Verburgt, Y. Zhang, Z. Zhang, H. Fujuta, M. Sekijima, D. Kihara, O. Khan, S. Kotelnikov, U. Ghani, D. Padhorny, D. Beglov, S. Vajda, D. Kozakov, S. Negi S, T. Ricciardelli, D. Barradas-Bautista, Z. Cao, M. Chawla, L. Cavallo, R. Oliva, R. Yin, M. Cheung, J. Guest, J. Lee, B. Pierce, B. Shor, T. Cohen, M. Halfon, D. Schneidman-Duhovny, S. Zhu, R. Yin, Y. Sun, Y. Shen, M. Maszota-Zieleniak, K. Bojarski K, E. Lubecka, M. Marcisz, A. Danielsson, L. Dziadek, M. Gaardlos, A. Giełdoń, J. Liwo, S. Samsonov, R. Slusarz, K. Zieba, A. Sieradzan, C. Czaplewski , S. Kobayashi, Y. Miyakawa, Y. Kiyota, M. Takeda-Shitaka, K. Olechnovič, L. Valančauskas, J. Dapkūnas, C. Venclovas, B. Wallner, L. Yang, C. Hou, X. He, S. Guo, S. Jiang, X. Ma, R. Duan, L. Qiu, X. Xu, X. Zou, S. Velankar, S. Wodak. [Impact of AlphaFold on Structure Prediction of Protein Complexes: The CASP15-CAPRI Experiment](https://doi.org/10.1002/prot.26609) _Proteins: Struc. Funct. &amp; Bioinformatics_ *12*, 1658-1683 (2023).

* R. Shukla, A.J. Peoples, K.C. Ludwig, S. Maity, M.G.N. Derks, S. de Benedetti, A.M. Krueger, B.J.A. Vermeulen, F. Lavore, R.V. Honorato, F. Grein, **A.M.J.J. Bonvin**, U. Kubitscheck, E. Breukink, C. Achorn, A. Nitti, C.J. Schwalen, A.L. Spoering, L.L. Ling, D. Hughes, M. Lelli, W.H. Roos, K. Lewis, T. Schneider, M. Weingarth. [A new antibiotic from an uncultured bacterium binds to an immutable target](https://doi.org/10.1016/j.cell.2023.07.038). _Cell_  *186*, 4059-4073, DOI:10.1016/j.cell.2023.07.038 (2023).

* F. van der Krift, D.W. Zijlmans, R. Shukla, A. Javed, P.I. Koukos, L.L.E. Schwarz, E.P.M. Timmermans-Sprang, P.E.M. Maas, D. Gahtory, M. van den Nieuwboer, J.A. Mol, G.J. Strous, **A.M.J.J. Bonvin**, M. van der Stelt, E.J.A. Veldhuizen, M.s Weingarth, M. Vermeulen, J. Klumperman, M.M. Maurice. [A novel antifolate suppresses growth of FPGS-deficient cells and overcomes methotrexate resistance](https://doi.org/10.26508/lsa.202302058). _Life Science Alliance_ *6* e202302058; DOI: 10.26508/lsa.202302058 (2023).

* S. van Keulen and **A.M.J.J. Bonvin**. [Improving the Quality of Co-evolution Intermolecular Contact Prediction with DisVis](http://doi.org/10.1002/prot.26514). _Proteins: Struc. Funct. &amp; Bioinformatics_ *91*, 1407-1416 (2023). Preprint [here](https://doi.org/10.1101/2022.11.03.515006).

* H. Schweke, Q. Xu, G. Tauriello, L. Pantolini, T. Schwede, F. Cazals, A. Lhéritier, J. Fernandez-Recio, L.A. Rodríguez-Lumbreras, O. Schueler-Furman, J.K. Varga, B. Jiménez-García, M.F. Réau, **A.M.J.J. Bonvin**, C. Savojardo, P.-L. Martelli, R. Casadio, J. Tubiana, H. Wolfson, R. Oliva, D. Barradas-Bautista, T. Ricciardelli, L. Cavallo, Č. Venclovas, K. Olechnovič, R. Guerois, J. Andreani, J. Martin, X. Wang, D. Kihara , A. Marchand, B. Correia, X. Zou, S. Dey, R. Dunbrack, E. Levy, S. Wodak. [Discriminating physiological from non-physiological interfaces in structures of protein complexes: a community-wide study](https://doi.org/10.1002/pmic.202200323). _Proteomics_ *23*, 1-21 (2023).

* M.F. Réau, N.Renaud, L.C. Xue and **A.M.J.J. Bonvin**. [DeepRank-GNN: A Graph Neural Network Framework to Learn Patterns in Protein-Protein Interfaces](https://doi.org/10.1093/bioinformatics/btac759). _Bioinformatics_ *39*, btac759, 1-8 (2023). Preprint [here](https://doi.org/10.1101/2021.12.08.471762).

* F. Ambrosetti, Z. Jandova and **A.M.J.J. Bonvin**. [A protocol for information-driven antibody-antigen modelling with the HADDOCK2.4 webserver](https://link.springer.com/protocol/10.1007/978-1-0716-2609-2_14). _Methods in molecular biology_ *2552*, 267-282 (2023). [Preprint](https://arxiv.org/abs/2005.03283).

* Y. Jung, C. Geng, **A.M.J.J. Bonvin**, L.C. Xue, V.G. Honavar. [MetaScore: A novel machine-learning based approach to improve traditional scoring functions for scoring protein-protein docking conformations](https://doi.org/10.3390/biom13010121). _Biomolecules_, *13*, 121 (1-200) (2023). Preprint [here](https://doi.org/10.1101/2021.10.06.463442)).


## 2022

* M.J. Abraham, R. Apostolov, P. Bauer, **A.M.J.J. Bonvin**, J.M. Correia Teixeira, B.L. de Groot, V. Gapsys, G. Groenhof, B. Hess, E. Laure, E. Lindahl, A.S.J. Melquiond, D. Morozov, J.P.G.L.M. Rodrigues, M. Trellet, R. Vargas Honorato. [BioExcel Updated Whitepaper on Scientific Software Development](https://doi.org/10.5281/zenodo.6404474). _Zenodo_ https://doi.org/10.5281/zenodo.6404474 (2022).

* M. Stahl Madsen, M. Broekema, M. Rønn Madsen, A. Koppen, A. Borgman, C. Gräwe, E. Thomsen, D. Westland, M. Kranendonk, M. Groot Koerkamp, N. Hamers, **A.M.J.J. Bonvin**, J. Ramos Pittol, K. Natarajan, S. Kersten, F. Holstege, H. Monajemi, S. van Mil, M. Vermeulen, B. Kragelund, D. Cassiman, S. Mandrup and E. Kalkhoven. [PPARγ lipodystrophy mutants reveal intermolecular interactions required for enhancer activation](https://doi.org/10.1038/s41467-022-34766-9). _Nature Comm._,  *13*, 7090 (2022).

* A. Basciu, L. Callea, S. Motta, **A.M.J.J. Bonvin**, L. Bonati and A.V. Vargiu. [No dance, no partner! A tale of receptor flexibility in docking and virtual screening](https://doi.org/10.1016/bs.armc.2022.08.006). _Annual Reports in Medicinal Chemistry_ (2022)

* R. Shukla, F. Lavore, So. Maity, M.G.N. Derks, C.R. Jones4, B.J.A. Vermeulen, A. Melcrová, M.A. Morris, L.M. Becker, X. Wang, R. Kumar, J. Medeiros-Silva, R.A.M. van Beekveld, **A.M.J.J. Bonvin**, J. Lorent, M. Lelli, J. Nowick, H.D. MacGillavry, A.J. Peoples, A.L. Spoering, L.L. Ling, D.E. Hughes, W.H. Roos, E. Breukink, K. Lewis and M. Weingarth. [Teixobactin kills bacteria by a two-pronged attack on the cell envelope](https://www.nature.com/articles/s41586-022-05019-y). _Nature_ *608*, 390-396 (2022)

* C.J. Buchanan, B. Gaunt, P.J. Harrison, Y. Yang, J. Liu, A. Khan, A.M. Giltrap, A. Le Bas, P.N. Ward, K. Gupta, M. Dumoux, T.K. Tan, L. Schimaski, S. Daga, N. Picchiotti, M. Baldassarri, E. Benetti, C. Fallerini, F. Fava, A. Giliberti, P.I. Koukos, M.J. Davy, A. Lakshminarayanan, X. Xue, G. Papadakis, L.P. Deimel, V. Casablancas-Antràs, T.D.W. Claridge, **A.M.J.J. Bonvin**, Q.J. Sattentau, S. Furini, M. Gori, J.Huo, R.J. Owens, C. Schaffitzel, I. Berger, A. Renieri, GEN-COVID Multicenter Study, J.H. Naismith, A. Baldwin, B.G. Davis. [Pathogen-sugar interactions revealed by universal saturation transfer analysis](https://doi.org/10.1126/science.abm3125). _Science_ *377*, 385(1-17) (2022).

* V. Charitou, S.C. van Keulen and **A.M.J.J. Bonvin**. [A Cyclisation and Docking Protocol for Cyclic Peptide-Protein Modelling using HADDOCK2.4](https://doi.org/10.1021/acs.jctc.2c00075). _J. Chem. Theory Comput._ *18*, 4027–4040 (2022).

* B. Garofalo, **A.M.J.J. Bonvin**, A. Bosin, F.P. Di Giorgio, R. Ombrato and A.V. Vargiu. [Molecular Insights Into Binding and Activation of the Human KCNQ2 Channel by Retigabine](https://doi.org/10.3389/fmolb.2022.839249). _Frontiers Mol. Biosci._ *9*, 839249 (2022).

* N. Rosário-Ferreir, **A.M.J.J. Bonvin** and Irina S. Moreira. [Using machine-learning-driven approaches to boost hot-spot's knowledge.](http://doi.org/10.1002/wcms.1602) _WIREs Comp. Mol. Sci._, *12*, e1602 (2022).

* T Neijenhuis, S.C. van Keulen and **A.M.J.J. Bonvin**. [Interface Refinement of Low-to-Medium Resolution Cryo-EM Complexes using HADDOCK2.4](https://doi.org/10.1016/j.str.2022.02.001). _Structure_ *30*, 476-484 (2022).


## 2021

* G. Dudas, S. Hong, B. Potter, S. Calvignac-Spencer, F. Niatou-Singa, T. Tombolomako, T. Fuh-Neba, U. Vickos, M. Ulrich, F. Leendertz, K. Khan, C. Huber, A. Watts, I. Olendraitė, J. Snijder, K. Wijnant, **A.M.J.J. Bonvin**, P. Martres, S. Behillil, A. Ayouba, M. Maidadi, D. Djomsi, C. Godwe, C. Butel, A. Šimaitis, M. Gabrielaite, M. Katėnaitė, R. Norvilas, L. Raugaitė, G. Koyaweda, J. Kaleb Kandou, R. Jonikas, I. Nasvytienė, Ž. Žemeckienė, D. Gečys, K. Tamušauskaitė, M. Norkienė, E. Vasiliūnaitė, D. Žiogienė, A. Timinskas, M. Šukys, M. Šarauskas, G. Alzbutas, A. Amuri, E. Lusamaki, J. Makangara, F. Muyembe, E. Lofiko, P. Mbala, J. Muyembe-Tamfum, M. Darnycka Belizaire, R. Essomba, M. Claire Assoumou, A. Mboringong, A. Baba Dieng, D. Juozapaitė, S. Hosch, J. Obama, M. Ondo'o Ayekaba, D. Naumovas, A. Pautienius, C. Rafaï, A. Vitkauskienė, R. Ugenskienė, A. Gedvilaitė, D. Čereškevičius, V. Lesauskaitė, L. Žemaitis, L. Griškevičius and G. Baele. [Emergence and spread of SARS-CoV-2 lineage B.1.620 with variant of concern-like mutations and deletion](https://doi.org/10.1038/s41467-021-26055-8). _Nature Comm._ *12*, 5769 (2021). Preprint available [here](https://www.medrxiv.org/content/10.1101/2021.05.04.21256637v1).

* M.F. Lensink, G. Brysbaert, T. Mauri, N. Nadzirin, S. Velankar, R.A.G. Chaleil, T. Clarence, P.A. Bates, R. Kong, B. Liu, G. Yang, M. Liu, H. Shi, X. Lu, S. Chang, R.S. Roy, F. Quadir, J. Liu, J. Cheng, A. Antoniak, C. Czaplewski, A. Giełdoń, M. Kogut, A.G. Lipska, A. Liwo, E.A. Lubecka, M. Maszota-Zieleniak, A.K. Sieradzan, R. Ślusarz, P.A. Wesołowski, K. Zięba, C.A. Del Carpio Muñoz, E. Ichiishi, A. Harmalkar, J.J. Gray, **A.M.J.J. Bonvin**, F. Ambrosetti, R. Vargas Honorato, Z. Jandova, B. Jiménez-García, P.I. Koukos, S. Van Keulen, C.W. Van Noort, M. Réau, J. Roel-Touris, S. Kotelnikov, D. Padhorny, K.A. Porter, A. Alekseenko, M. Ignatov, I. Desta, R. Ashizawa, Z. Sun, U. Ghani, N. Hashemi, S. Vajda, D. Kozakov, M. Rosell, L.A. Rodríguez-Lumbreras, J. Fernandez-Recio, A. Karczynska, S. Grudinin, Y. Yan, H. Li, P. Lin, S. Huang, C. Christoffer, G. Terashi, J. Verburgt, D. Sarkar, T. Aderinwale, X. Wang, D. Kihara, T. Nakamura, Y. Hanazono, R. Gowthaman, J.D. Guest, R. Yin, G. Taherzadeh, B.G. Pierce, D. Barradas-Bautista, Z. Cao, L. Cavallo, R. Oliva, Y. Sun, S. Zhu, Y. Shen, T. Park, H. Woo, J. Yang, S. Kwon, J. Won, C. Seok, Y. Kiyota, S. Kobayashi, Y. Harada, M. Takeda-Shitaka, P.J. Kundrotas, A. Singh, I.A. Vakser, J. Dapkūnas, . Olechnovič, Č. Venclovas, R. Duan, L. Qiu, X. Xu, S. Zhang, X. Zou, S.J. Wodak. [Prediction of protein assemblies, the next frontier: The CASP14-CAPRI experiment](http://doi.org/10.1002/prot.26222). _Proteins: Struc. Funct. &amp; Bioinformatics_, *89*, 1800-1823 (2021).

* R. Vargas Honorato, P.I. Koukos, B. Jimenez-Garcia, A. Tsaregorodtsev, M. Verlato, A. Giachetti, A. Rosato and **A.M.J.J. Bonvin**. [Structural biology in the clouds: The WeNMR-EOSC Ecosystem](https://doi.org/10.3389/fmolb.2021.729513). _Frontiers Mol. Biosci._ *8*, fmolb.2021.729513 (2021).

* P.I. Koukos, M.F. Reau and **A.M.J.J. Bonvin**. [Shape-restrained modelling of protein-small molecule complexes with HADDOCK](https://doi.org/10.1021/acs.jcim.1c00796). _J. Chem. Inf. and Mod._ *61*, 4807–4818 (2021). BioRxiv preprint available [here](https://doi.org/10.1101/2021.06.10.447890).

* **A.M.J.J. Bonvin**. [50 years of PDB: A catalyst in structural biology](https://doi.org/10.1038/s41592-021-01138-y). _Nature Methods_ *18*, 448–449 (2021).

* U.B. le Paige, S. Xiang, M.M.R.M. Hendrix, Y. Zhang., G.E. Folkers, M. Weingarth, **A.M.J.J. Bonvin**, T.G. Kutateladze, I.K.Voets, M. Baldus and H. van Ingen. [Characterization of nucleosome sediments for protein interaction studies by solid-state NMR spectroscopy](https://mr.copernicus.org/articles/2/105/2021/). _Magn. Reson._, *2*, 105-116 (2021).

* Z. Jandova, A.V. Vargiu and **A.M.J.J. Bonvin**. [Native or non-native protein-protein docking models? Molecular dynamics to the rescue](https://doi.org/10.1021/acs.jctc.1c00336). _J. Chem. Theo. and Comp._  *17*, 5944−5954 (2021).

* C.W. van Noort, R.V. Honorato and **A.M.J.J. Bonvin**. [Information-Driven Modeling of Biomolecular Complexes](https://doi.org/10.1016/j.sbi.2021.05.003).  _Curr. Opin. Struct. Biol._ *70*, 70-77 (2021).

* P. Matos-Filipe, A.J. Preto, P.I. Koukos, J. Mourão, **A.M.J.J. Bonvin** and I.S. Moreira. [MENSADB: A Thorough Structural Analysis of Membrane Protein Dimers](https://doi.org/10.1093/database/baab013). _Database_, doi:10.1093/database/baab013, 1-10 (2021).

* N.Renaud, C. Geng, S. Georgievska, F. Ambrosetti, L. Ridder, D.F Marzella, M.F. Réau, **A.M.J.J. Bonvin** and L.C. Xue. [DeepRank: A deep learning framework for data mining 3D protein-protein interfaces](https://doi.org/10.1038/s41467-021-27396-0). _Nature Comm._ *12*, 7068 (2021).

* S. Nizamuddin, S. Koidl, T. Bhuiyan, T.V. Werner, M.L. Biniossek, **A.M.J.J. Bonvin**, S. Lassmann and H.Th.M. Timmers. [Integrating quantitative proteomics with accurate genome profiling of transcription factors by greenCUT&RUN](https://doi.org/10.1093/nar/gkab038). _Nucl. Acids Res._ *49*, e49-‌e49 (2021).

* O.M.H. Salo-Ahen#, I. Alanko, R. Bhadane, **A.M.J.J. Bonvin**#, R. Vargas Honorato, S. Hossain, A.H. Juffer, A. Kabedev, M. Lahtela-Kakkonen, A. Støttrup Larsen, E. Lescrinier, P. Marimuthu, M.. Mirza, G. Mustafa, A. Nunes-Alves#, T. Pantsar, A. Saadabadi, K. Singaravelu, M. Vanmeert. [Molecular dynamics simulations in drug discovery and pharmaceutical development](https://doi.org/10.3390/pr9010071 ). _Processes_. *9*, 71 (2021).

* B. Jiménez-García, J.M.C. Teixeira, M.E. Trellet, J.P.G.L.M. Rodrigues and **A.M.J.J. Bonvin**. [PDB-Tools Web: A user-friendly interface for the manipulation of PDB files](http://dx.doi.org/10.1002/prot.26018) _Proteins: Struc. Funct. &amp; Bioinformatics_, *89*, 330-335 (2021).


## 2020

* A. Saponaro, V. Maione, **A.M.J.J. Bonvin** and F. Cantini. [Understanding Docking Complexes of Macromolecules Using HADDOCK: The Synergy between Experimental Data and Computations](https://doi.org/10.21769/BioProtoc.3793). _Bio-protocol_ *10*, e3793 (2020).

* M. Schwarz, P. Skrinjar, M.J. Fink, S. Kronister, T. Mechtler, P.I. Koukos, **A.M.J.J. Bonvin**, D. Kasper and H. Mikula. [A click-flipped enzyme substrate boosts the performance of the diagnostic screening for Hunter syndrome](https://doi.org/10.1039/D0SC04696E). _Chem. Sci._, 10.1039/D0SC04696E (2020).

* C.Y.L. Lau, F, Fontana, L, Mandemaker, D, Wezendonk, B, Vermeer, **A.M.J.J. Bonvin**, R. de Vries, H. Zhang, K. Remaut, J. van den Dikkenberg, J. Silva, A. Hassan, B. Perrone, R. Kuemmerle, F. Gelain, W. Hennink, M. Weingarth, and E. Mastrobattista. [Control over the fibrillization yield by varying the oligomeric nucleation propensities of self-assembling peptides](https://doi.org/10.1038/s42004-020-00417-7). _Comm. Chem._ *3*, 164 (2020).

* V. Tozzini , G. Palermo , **A.M.J.J. Bonvin** , M. Dal Peraro and R.E. Amaro. [Editorial: Multiscale Modeling from Macromolecules to Cell: Opportunities and Challenges of Biomolecular Simulations](https://www.frontiersin.org/articles/10.3389/fmolb.2020.00194/full). _Front. Mol. Biosci._ 10.3389/fmolb.2020.00194 (2020).

* J.L. Roel-Touris,  B. Jiménez-García and **A.M.J.J. Bonvin**. [Integrative Modeling of Membrane-associated Protein Assemblies](https://doi.org/10.1038/s41467-020-20076-5). _Nature Comm._, *11*, 6210 (2020).
 
* H.H. Rabouw, L.J. Visser, T.C. Passchier, M.A. Langereis, F. Liu, P. Giansanti, A.L.W. van Vliet, J.G. Dekker, S.G. van der Grein, J.G. Saucedo, A.A. Anand, M.E. Trellet, **A.M.J.J. Bonvin**, P. Walter, A.J.R. Heck, R.J. de Groot and F.J.M. van Kuppeveld. [Inhibition of the integrated stress response by viral proteins that block p-eIF2–eIF2B association](https://doi.org/10.1038/s41564-020-0759-0). _Nature Microbiology_, *58*, 1361-1373 (2020).

* A. Leitner, **A.M.J.J. Bonvin**, C.H. Borchers, R.J. Chalkley, . Chamot-Rooke, C.W. Combe, J. Cox, M. Dong, L. Fischer, M. Götze, F.C. Gozzo, A.J.R. Heck, M.R. Hoopmann, L. Huang, Y. Ishihama, A.R. Jones, N. Kalisman, O. Kohlbacher, K. Mechtler, R.L. Moritz, E. Netz, P.r Novak, E. Petrotchenko, A. Sali, R.A. Scheltema, C. Schmidt, D. Schriemer, A. Sinz, F. Sobott, F. Stengel, K. Thalassinos, H. Urlaub, R. Viner, J.A. Vizcaino, M. R. Wilkins, J. Rappsilber. [Towards Increased Reliability, Transparency and Accessibility in Crosslinking Mass Spectrometry](https://doi.org/10.1016/j.str.2020.09.011) _Structure_, _28_, 1259-1268 (2020).

* J. Roel-Touris and **A.M.J.J. Bonvin**. [Coarse-Grained (Hybrid) Integrative Modeling of Biomolecular Interactions](https://doi.org/10.1016/j.csbj.2020.05.002). _Comp. Struct. Biotech. J._, *18*, 1182-1190 (2020).

* R. Shukla, J. Medeiros-Silva, A. Parmar, B.J.A. Vermeulen, S. Das, A. Paioni, S. Jekhmane, J. Lorent, **A.M.J.J. Bonvin**, M. Baldus, M. Lelli, E.J.A. Veldhuizen, E. Breukink, I. Singh and M. Weingarth. [Mode of action of teixobactins in cellular conditions](https://www.nature.com/articles/s41467-020-16600-2). _Nature Communications_, *11*, 2848 (2020).

* C. Orengo, S. Velankar, S. Wodak, V. Zoete, **A.M.J.J. Bonvin**, A. Elofsson, K.A Feen.stra, D.L. Gerloff, T. Hamelryck, J.M. Hancock, M. Helmer-Citterich, A. Hospital, M. Orozco, A. Perrakis, M. Rarey, C. Soares, J.L. Sussman, J.M. Thornton, P. Tuffery, G. Tusnady, R. Wierenga, T. Salminen, B. Schneider. [A community proposal to integrate structural bioinformatics activities in ELIXIR (3D-Bioinfo Community)](https://f1000research.com/articles/9-278/v1). _F1000 Research_, Awaiting peer review (2020).

* N. Renault, Y. Jung, V. Honavar, C. Geng, **A.M.J.J. Bonvin**  and L. Xue. [iScore: an MPI supported software for ranking protein-protein docking models based on a random walk graph kernel and support vector machines](https://doi.org/10.1016/j.softx.2020.100462). _SoftwareX_, *11*, 100462 (2020).

* F. Ambrosetti, T.H. Olsed, P.P. Olimpieri, B. Jiménez-García, E. Milanetti, P. Marcatilli and **A.M.J.J. Bonvin**. [proABC-2: PRediction Of AntiBody Contacts v2 and its application to information-driven docking](https://doi.org/10.1093/bioinformatics/btaa644). _Bioinformatics_, *36*, 5107–5108 (2020).

* C. van Leeuwen, D. Podareanu, V. Codreanu, M. Cai, A. Berg, S. Portegies Zwart, R. Stoffer, M. Veerman, C. van Heerwaarden, S. Otten, S. Caron, C. Geng, F. Ambrosetti, A.M.J.J. Bonvin. [Deep-learning enhancement of large scale numerical simulations](https://arxiv.org/abs/2004.03454). _Whitepaper_ (2020).

* K. Elez, **A.M.J.J. Bonvin**#, Anna Vangone#. [Biological vs crystallographic protein interfaces: An overview of computational approaches for their classification](https://doi.org/10.3390/cryst10020114 ). _Crystals_, *10*, 114 (2020).

* P.I. Koukos, J. Roel-Touris, F. Ambrosetti, C. Geng, J. Schaarschmidt, M.E. Trellet, A.S.J. Melquiond, L.C. Xue, R.V. Honorato, I. Moreira, Z. Kurkcuoglu, A. Vangone and **A.M.J.J. Bonvin**. [An overview of data-driven HADDOCK strategies in CAPRI rounds 38-45](https://doi.org/10.1002/prot.25869). _Proteins: Struc. Funct. &amp; Bioinformatics_, *88*, 1029-1036 (2020).

* F. Ambrosetti, B. Jiménez-García, J. Roel-Touris and **A.M.J.J. Bonvin**. [Modeling Antibody-Antigen Complexes by Information-Driven Docking](https://doi.org/10.1016/j.str.2019.10.011). _Structure_, *28*, 119-129 (2020). Preprint freely available from [here](https://papers.ssrn.com/sol3/papers.cfm?abstract_id=3362436).

* Z. Kurkcuoglu and **A.M.J.J. Bonvin**. [Pre- and post-docking sampling of conformational changes using ClustENM and HADDOCK for protein-protein and protein-DNA systems](https://doi.org/10.1002/prot.25802). _Proteins: Struc. Funct. &amp; Bioinformatics_, *88*, 292-306 (2020).

* T. Lawson, S. El-Kamand, D. Boucher, D. Cong Duong, R. Kariawasam, **A.M.J.J. Bonvin**, D.J. Richard, R. Gamsjaeger1 and Liza Cubeddu. [The structural details of the interaction of single-stranded DNA binding protein hSSB2 (NABP1/OBFC2A) with UV-damaged DNA](https://doi.org/10.1002/prot.25806). _Proteins: Struc. Funct. &amp; Bioinformatics_, *88*, 319-326 (2020).

* A. Basciu, P.I. Koukos, G. Malloci, **A.M.J.J. Bonvin** and A.V. Vargiu. [Coupling enhanced sampling of the apo‐receptor with template‐based ligand conformers selection: performance in pose prediction in the D3R Grand Challenge 4](https://doi.org/10.1007/s10822-019-00244-6). _J. Comp. Aid. Mol. Des._ *34*, 149-162 (2020). A preprint can be downloaded from [here](https://arxiv.org/abs/2005.04142).

* J.L. Roel Touris, **A.M.J.J. Bonvin** and B. Jimenez-Garcia. [LightDock goes information-driven](https://doi.org/10.1093/bioinformatics/btz642). _Bioinformatics_, *36*, 950–952 (2020).

* M.E. Trellet, G. van Zundert and **A.M.J.J. Bonvin**. [Protein-protein modelling using cryo-EM restraints](https://dx.doi.org/10.1007/978-1-0716-0270-6_11). In:  _Structural Bioinformatics. Methods in Molecular Biology_, vol 2112. Humana, New York, NY, (2020). A preprint is available [here](https://arxiv.org/abs/2005.00435).

* P.I. Koukos and **A.M.J.J. Bonvin**. [Integrative modelling of biomolecular complexes](https://doi.org/10.1016/j.jmb.2019.11.009). _J. Mol.  Biol._ *432*, 2861-2881 (2020)

* C. Geng,  Y. Jung, N. Renaud,  V. Honavar, **A.M.J.J. Bonvin** and L.C. Xue. [iScore: A novel graph kernel-based function for scoring protein-protein docking models](https://doi.org/10.1093/bioinformatics/btz496) _Bioinformatics_ *36*, 112–121 (2020).


## 2019 

* M.F. Lensink, G. Brysbaert  N. Nadzirin, S. Velankar, R.A.G. Chaleil,  T, Gerguri,  P.A. Bates,  E. Laine,  A. Carbone,  S. Grudinin,  R. Kong,  R. Liu, X. Xu,  H, Shi, S. Chang,  M. Eisenstein,  A.a Karczynska,  C. Czaplewski, E. Lubecka, A. Lipska, P. Krupa, M. Mozolewska, L. Golon, S. Samsonov, A. Liwo, S. Crivelli, G. Pagès, M. Karasikov, M. Kadukova, Y. Yan, S. Huang, M. Rosell, L.A. Rodríguez‐Lumbreras, M. Romero‐Durana, L. Díaz‐Bueno, J. Fernandez‐Recio, C. Christoffer, G. Terashi, W. Shin, T. Aderinwale, S. Raghavendra Maddhuri Venkata Subraman, D. Kihara, D. Kozakov, S. Vajda, K. Porter, D. Padhorny, I. Desta, D. Beglov, M. Ignatov, S. Kotelnikov, I.H. Moal, D.W. Ritchie, I. Chauvot de Beauchêne, B. Maigret, M. Devignes, M.E. Ruiz Echartea, D, Barradas‐Bautista, Z. Cao, L. Cavallo, R. Oliva, Y. Cao, Y. Shen, M. Baek, T. Park, H. Woo, C. Seok, M. Braitbard, L. Bitton, D. Scheidman‐Duhovny, J. Dapkūnas, K. Olechnovič, Č. Venclovas, P.J. Kundrotas, S. Belkin, D. Chakravarty, V.D. Badal, I.A. Vakser, T.Vreven, S. Vangaveti, T. Borrman, Z. Weng, J.D. Guest, R. Gowthaman, B.G. Pierce, X. Xu, R. Duan, L. Qiu, J. Hou, B. Ryan Merideth, Z. Ma, J. Cheng, X. Zou, P.I. Koukos, J. Roel‐Touris, . Ambrosetti, C. Geng. J. Schaarschmidt, M.E. Trellet, A.S.J. Melquiond, L. Xue, B. Jiménez‐Garcí, C.W. van Noort, R.V. Honorato, **A.M.J.J. Bonvin**, S.J. Wodak. [Blind prediction of homo‐ and hetero‐protein complexes: The CASP13‐CAPRI experiment](https://doi.org/10.1002/prot.25838).
_Proteins: Struc. Funct. &amp; Bioinformatics_, *87*, 1200-1221 (2019)

* H.M. Berman, P,D. Adams, **A.M.J.J. Bonvin**, S.K. Burley, B. Carragher, W. Chiu, F. DiMaio, T.E. Ferrin, M.J. Gabanyi, T.D. Goddard, P.R. Griffin, J. Haas, C.A. Hanke, J.C. Hoch, G. Hummer, G. Kurisu, C.L. Lawson, A. Leitner, J.L. Markley, J. Meiler, G.T. Montelione, G.N. Phillips Jr., T. Prisner, J. Rappsilber, D.C. Schriemer, T. Schwede, C.A.M. Seidel, T.S. Strutzenberg, D.I. Svergun, E. Tajkhorshid, J. Trewhella, B. Vallat, S. Velankar, G.W. Vuister, B. Webb, J.D. Westbrook, K.L. White, A. Sali. [Federating Sructural Models and Data: Outcomes from a Workshop on Archiving Integrative Structures](https://doi.org/10.1016/j.str.2019.11.002). _Structure_ *27*, 1745-1759 (2019).

* M. Abraham, R. Apostolov, J. Barnoud, P. Bauer, C. Blau, **A.M.J.J. Bonvin**, M. Chavent, J. Chodera, K. Čondić-Jurkić, L. Delemotte, H. Grubmüller, R. Howard, E.J. Jordan, E. Lindal, O.H.S. Ollila, J. Selent, D. Smith, P. Stansfeld, J. Tiemann, M. Trellet, C. Woods and A. Zhmurov. [Sharing Data from Molecular Simulations](https://doi.org/10.1021/acs.jcim.9b00665). _J. Chem. Inf. and Model._ *59*, 4093-4099 (2019).

* R.V. Honorato, J. Roel-Touris and **A.M.J.J. Bonvin**. [MARTINI-based protein-DNA coarse-grained HADDOCKing](https://doi.org/10.3389/fmolb.2019.00102). _Frontiers in Molecular Biosciences_, *6*, 102 (2019).

* J. Roel-Touris, C.G. Don, R.V. Honorato, J.P.G.L.M Rodrigues and **A.M.J.J. Bonvin**. [Less is more: Coarse-grained integrative modeling of large biomolecular assemblies with HADDOCK](https://doi.org/10.1021/acs.jctc.9b00310). _J. Chem. Theo. and Comp._, *15*, 6358-6367 (2019).

* R.A. Norman, F. Ambrosetti, **A.M.J.J. Bonvin**, L.J. Colwell, S. Kelm, S. Kumar and K. Krawczyk. [Computational approaches to therapeutic antibody design: established methods and emerging trends](https://doi.org/10.1093/bib/bbz095). _Briefings in Bioinformatics_, bbz095 (2019).

* L. Goldbach, B. Vermeulen, S. Caner, M. Liu, C.Tysoe, L. van Gijzel, R. Yoshisada, M.E. Trellet, H. van Ingen, G. Brayer, **A.M.J.J. Bonvin** and S. Jongkees. [Folding Then Binding vs. Folding Through Binding in Macrocyclic Peptide Inhibitors of Human Pancreatic alpha-Amylase](https://pubs.acs.org/doi/abs/10.1021/acschembio.9b00290). _ACS Chemical Biology_ *14*, 1751-1759 (2019).

* B. Jiménez-García, K. Elez, P.I. Koukos, **A.M.J.J. Bonvin** and A. Vangone. [PRODIGY-crystal: a web-tool for classification of biological interfaces in protein complexes](https://doi.org/10.1093/bioinformatics/btz437). _Bioinformatics_, *35*, 4821–4823 (2019).

* A. Basciu,  G. Malloci,  F. Pietrucci,  **A.M.J.J. Bonvin** and A.V. Vargiu.
[Holo-like and druggable protein conformations from enhanced-sampling of binding pocket shape](https://dx.doi.org/10.1021/acs.jcim.8b00730). _J. Chem. Inf. and Mod._, *59*, 1515–1528 (2019).

* C. Morris, P. Andreetto, L. Banci, **A.M.J.J. Bonvin**, G.G. Chojnowski, L. del Cano, J.M. Carazo; P. Conesa, S. Daenke, G. Damaskos, A. Giachetti, N. Haley, M. Hekkelman, P. Heuser, R. Joosten, D. Kouřil, A. Křenek, T. Kulhánek, V. Lamzin, N. Nadzirin, A. Perrakis, A. Rosato, F. Sanderson, J. Segura, J. Schaarschmidt, E. Sobolev, S. Traldi, M.E. Trellet, S, Velankar, M. Verlato, and M. Winn. 
[West-Life: A Virtual Research Environment for structural biology](https://doi.org/10.1016/j.yjsbx.2019.100006).
_Journal of Structural Biology: X_, *1*, 100006 (2019).

* C. Geng. Li  J. Roel-Touris and **A.M.J.J. Bonvin**. [Finding the ∆∆G spot: Are predictors of binding affinity changes upon mutations in protein-protein interactions ready for it?](https://onlinelibrary.wiley.com/doi/full/10.1002/wcms.1410) _WIREs Computational Molecular Science_ *9*, e1410 (2019).

* C. Geng, A. Vangone, G.E. Folkers, L.C. Xue and **A.M.J.J. Bonvin**. [iSEE: Interface Structure, Evolution and Energy-based machine learning predictor of binding affinity changes upon mutations](https://doi.org//10.1002/prot.25630).
  _Proteins: Struc. Funct. &amp; Bioinformatics_ *87*, 110-119  (2019).

* P.I. Koukos, L.C. Xue and **A.M.J.J. Bonvin**. [Protein-ligand pose and affinity prediction. Lessons from D3R Grand Challenge 3](https://doi.org/10.1007/s10822-018-0148-4).  _J. Comp. Aid. Mol. Des._ *33*, 83-91 (2019).

* M.F. Broekema, M.P.G. Massink, C. Donato, J. de Ligt, J. Schaarschmidt, M.G. Schooneman, F.S. van Nierop, D. Melchers, M.N. Gerding, R. Houtman, **A.M.J.J. Bonvin**, A.R. Majithia, H. Monajemi, G.W. van Haaften, M.R. Soeters and E. Kalkhoven. [Natural helix 9 mutants of PPARγ differently affect its transcriptional activity](https://doi.org/10.1016/j.molmet.2018.12.005). _Molecular Metabolism_ *20*, 115-127 (209).


## 2018 

* A. Vangone#, J. Schaarschmidt, P. Koukos, C. Geng, N. Citro, M.E. Trellet, L.C. Xue and **A.M.J.J. Bonvin**. [Large-scale prediction of binding affinity in protein-small ligand complexes: the PRODIGY-LIG web server](https://doi.org/10.1093/bioinformatics/bty816). _Bioinformatics_, *35*, 1585–1587 (2018).

* J.P.G.L.M. Rodrigues, J.M.C. Teixeira, M.E. Trellet and **A.M.J.J. Bonvin**. [pdb-tools: a swiss army knife for molecular structures](https://doi.org/10.12688/f1000research.17456.1). _F1000Research_, *7:1961* (2018)

* P.I. Koukos, I. Faro, C.W. van Noort and **A.M.J.J. Bonvin**. [A membrane protein complex docking benchmark](https://doi.org/10.1016/j.jmb.2018.11.005). _J. Mol. Biol._ *430*, 5246-‌5256 (2018).

* W. Roseboom, M.G. Nazir, N.Y. Meiresonne, T. Mohammadi, J. Verheul, H. Buncherd, **A.M.J.J. Bonvin**, L.J. de Koning, C.G. de Koster, L. de Jong*, T. Den Blaauwen*. [Mapping the contact sites of the Escherichia coli division initiating 
proteins FtsZ and ZapA by BAMG cross-linking and site directed mutagenesis](https://doi.org/10.3390/ijms19102928). _International Journal of Molecular Sciences_ *19*, 2928 (2018).

* D. Salomoni, I. Campos, L. Gaido, J. Marco de Lucas, P. Solagna, J. Gomes, L. Matyska, P. Fuhrman, M. Hardt, G. Donvito, L. Dutka, M. Plociennik, R. Barbera, I. Blanquer, A. Ceccanti, E. Cetinic, M. David, C. Duma, A. López-García, G. Moltó, P. Orviz, Z. Sustr, M. Viljoen, F. Aguilar, L. Alves, M. Antonacci, L. A. Antonelli, S. Bagnasco,  **A.M.J.J. Bonvin**, R. Bruno, Y. Chen, A. Costa, D. Davidovic, B. Ertl, M. Fargetta, S. Fiore, S. Gallozzi, Z. Kurkcuoglu, L. Lloret, J. Martins, A. Nuzzo, P. Nassisi, C. Palazzo, J. Pina, E. Sciacca, D. Spiga, M. Tangaro, M. Urbaniak, S. Vallero, B. Wegh, V. Zaccolo, F. Zambelli, T. Zok. [INDIGO-DataCloud: a Platform to Facilitate Seamless Access to E-Infrastructures](https://doi.org/10.1007/s10723-018-9453-3). _J. Grid. Comp._ *16*, 381-408 (2018).

* K. Elez, **A.M.J.J. Bonvin*** and A. Vangone. [Distinguishing crystallographic from biological interfaces in protein complexes: Role of intermolecular contacts and energetics for classification](https://doi.org/10.1186/s12859-018-2414-9). _BMC Bioinformatics_, *19 (Suppl 15)*, 438 (2018).

* **A.M.J.J. Bonvin**, E. Karaca, P.L. Kastritis & J.P.G.L.M. Rodrigues. Correspondence: [Defining distance restraints in HADDOCK](https://doi.org/10.1038/s41596-018-0017-6). _Nature Protocols_ *13*, 1503 (2018). [Free online only access](https://rdcu.be/1OyH)

* M.J. Abraham, A.S.J. Melquiond, E. Ippoliti, V. Gapsys, B. Hess, M. Trellet, J.P.G.L.M. Rodrigues, E. Laure, R. Apostolov, B.L. de Groot, **A.M.J.J. Bonvin**, E. Lindahl. [BioExcel Whitepaper on Scientific Software Development](https://doi.org/10.5281/zenodo.1194634). _Zenodo_ https://doi.org/10.5281/zenodo.1194634 (2018).

* **A.M.J.J. Bonvin**, C. Geng, M. van Dijk, E. Karaca, P. L. Kastritis, P.I. Koukos, Z. Kurkcuoglu, A.S.J. Melquiond, 
J.P.G.L.M. Rodrigues, J. Schaarschmidt, C. Schmitz, J. Roel-Touris, M.E. Trellet, S. de Vries, A. Vangone, L. Xue, G.C.P. van Zundert
[HADDOCK](https://doi.org/10.1007/978-3-642-35943-9_330-1). In _Encyclopedia of Biophysics_, In press (2018).

* A.J. Preto, J.G. Almeida, J. Schaarschmidt, L.C. Xue, I.S. Moreira and **A.M.J.J. Bonvin**. [Computational tools for the structural characterization of proteins and their complexes from sequence-evolutionary data](https://doi.org/10.1002/9780470027318.a9615). In: Roberts G., Watts A., European Biophysical Societies (eds) _Encyclopedia of Biophysics_. Springer, Berlin, Heidelberg (2018),

* S. Narasimhan, D. Mance, C. Pinto, M. Weingarth, **A.M.J.J. Bonvin** and Marc Baldus.
[Rapid Prediction of Multi-dimensional NMR Data Sets Using FANDAS](https://doi.org/10.1007/978-1-4939-7386-6_6). In: _Methods in Molecular Biology: Protein NMR_ Humana Press, New York, NY, 111-132 (2018).

* J. Schaarschmidt, B. Monastyrskyy, A.y Kryshtafovych and **A.M.J.J. Bonvin**. [Assessment of Contact Predictions in CASP12: Co-Evolution and Deep Learning Coming of Age](https://doi.org/10.1002/prot.25407). 
_Proteins: Struc. Funct. &amp; Bioinformatics_ *86(S1)*, 51-66 (2018).

* Z. Kurkcuoglu, P.I. Koukos, N. Citro, M.E. Trellet, J.P.G.L.M. Rodrigues, I.S. Moreira, J. Roel-Touris, A.S.J. Melquiond, C. Geng, J. Schaarschmidt, L.C. Xue, A. Vangone and **A.M.J.J. Bonvin**. [Performance of HADDOCK and a simple contact-based protein-ligand binding affinity predictor in the D3R Grand Challenge 2](https://doi.org/10.1007/s10822-017-0049-y). _J. Comp. Aid. Mol. Des._ *32*, 175-185 (2018).


## 2017 
<hr />
* A. Vangone and **A.M.J.J. Bonvin**. [PRODIGY: A Contact-based Predictor of Binding Affinity in Protein-protein Complexes](https://bio-protocol.org/e2124). _Bio-protocol_ *7*, e2124, DOI:10.21769/BioProtoc.2124 (2017)

* E. Karaca, J.P.G.L.M. Rodrigues, A. Graziadei, **A.M.J.J. Bonvin** and T. Carlomagno. [An Integrative Framework for Structure Determination of Molecular Machines](https://doi.org/10.1038/nmeth.4392). _Nature Methods_ *14*, 897-902 (2017).

* I.S. Moreira, P.I. Koukos, R. Melo, J.G. Almeida, A.J. Preto, J. Schaarschmidt, M. Trellet, Z.H. Gümüş, J. Costa and **A.M.J.J. Bonvin**. [SpotOn: High Accuracy Identification of Protein-Protein Interface Hot-Spots](https://doi.org/10.1038/s41598-017-08321-2). _Sci. Reports_. *7*:8007 (2017).

* J.G. Almeida, A.J. Preto, P.I. Koukos, **A.M.J.J. Bonvin** and I.S. Moreira. [Membrane proteins structures: A review on computational modeling tools](https://doi.org/10.1016/j.bbamem.2017.07.008). _Biochimica et Biophysica Acta (BBA) - Biomembranes_. *1859*, 2021-2039 (2017).

* K.M. Visscher, J. Medeiros-Silva, D. Mance, J.P.G.L.M. Rodrigues, M. Daniëls, **A.M.J.J. Bonvin**, M. Baldus, and M. Weingarth. [A study on the supramolecular organization of K+ channel clusters in membranes](https://doi.org/10.1002/anie.201705723). _Angew. Chem. Intl._ *56*, 13222-13227 (2017).

* A. Vangone, R. Oliva, L. Cavallo and **A.M.J.J. Bonvin**. [Prediction of Biomolecular Complexes](https://link.springer.com/chapter/10.1007/978-94-024-1069-3_8).
In: _From Protein Structure to Function with Bioinformatics._ Ed. Daniel John Rigden, Springer, 160-167 (2017).

* C. Geng, S. Narasimhan, J. P.G.L.M. Rodrigues and **A.M.J.J. Bonvin**.
[Information-driven, ensemble flexible peptide docking using HADDOCK](https://doi.org/10.1007/978-1-4939-6798-8_8).
_Methods in Molecular Biology: Modeling Peptide-Protein Interactions._ Eds Ora Schueler-Furman and Nir London. Humana Press Inc. *1561*, 109-138 (2017).

* R.C.G. de Bruin, A.G.M. Stam, A. Vangone, P. van Bergen en Henegouwen, H.M.W. Verheul, **A.M.J.J. Bonvin**, T.D. de Gruijl, H.J. van der Vliet.
[Prevention of Vγ9Vδ2-T cell activation by a Vγ9Vδ2-TCR nanobody](https://doi.org/10.4049/jimmunol.1600948).
_Journal of Immunology_, *198*, 308-317 (2017).

* T. Bohnuud, L. Luo, S.J. Wodak, **A.M.J.J. Bonvin**, Z. Weng, S. Vajda, O. Schueler-Furman and D. Kozakov.
[A benchmark testing ground for integrating homology modeling and protein docking](https://doi.org/10.1002/prot.25063).
_Proteins: Struc. Funct. &amp; Bioinformatics_, *85*, 10-17 (2017).

* G.C.P. van Zundert, M. Trellet, J. Schaarschmidt, Z. Kurkcuoglu, M. David, M. Verlato, A. Rosato and **A.M.J.J. Bonvin**. [The DisVis and PowerFit web servers: Explorative and Integrative Modeling of Biomolecular Complexes](https://doi.org/10.1016/j.jmb.2016.11.032).
_J. Mol. Biol._, *429*, 399-407 (2017).

## 2016
<hr />
* A. Vangone, J.P.G.L.M. Rodrigues, L.C. Xue, G.C.P. van Zundert, C. Geng, Z. Kurkcuoglu, M. Nellen, S. Narasimhan, E. Karaca, M. van Dijk, A.S.J. Melquiond, K.M. Visscher, M. Trellet, P.L. Kastritis and **A.M.J.J. Bonvin**.
[Sense and Simplicity in HADDOCK Scoring: Lessons from CASP-CAPRI Round 1](https://onlinelibrary.wiley.com/doi/10.1002/prot.25198/full).
_Proteins: Struc. Funct. &amp; Bioinformatics_, *85*, 417-423 (2016). See also erratum: [https://doi.org/10.1002/prot.25339](https://doi.org/10.1002/prot.25339)

* D. Walker, R. Grinter, I. Josts, K. Mosbahi, A. Roszak, R. Cogdell, **A.M.J.J. Bonvin**, J. Milner, S. Kelly, O. Byron and B. Smith.
[Structure of the bacterial plant-ferredoxin receptor FusA](https://doi.org/10.1038/ncomms13308).
_Nature Communications_, *7*, 13228 (2016).

* J. Felix, E. Kandiah, S. De Munck, Y. Bloch, G.C.P. van Zundert, K. Pauwels, A. Dansercoer, K. Novanska, R.J. Read, **A.M.J.J. Bonvin**, B. Vergauwen, K. Verstraete, I. Gutsche and S.N. Savvides.
[Structural basis of GM-CSF and IL-2 sequestration by the viral decoy receptor GIF to achieve immunomodulation](https://doi.org/10.1038/NCOMMS13228).
_Nature Communications_, *7*, 13228 (2016).

* Z. Kurkcuoglu and **A.M.J.J. Bonvin**. 
[Science in the clouds: virtualizing HADDOCK, PowerFit and DisVis using INDIGO-DataCloud solutions](https://f1000research.com/posters/5-874). _F1000Research_, *5*:874 (poster) (2016). 

* M. Trellet, J. Schaarschmidt, G.C.P. van Zundert, Z. Kurkcuoglu, A.S.J. Melquiond and **A.M.J.J. Bonvin**.
[Integrative modelling goes HPC/HTC](https://doi.org/10.7490/f1000research.1113051.1). _F1000Research_, *5*:2216 (poster) (2016).

* A.D. Spiliotopoulos, P.L. Kastritis, A.S.J. Melquiond, **A.M.J.J. Bonvin**, G. Musco, W. Rocchia and A. Spitaleri.
[dMM-PBSA: a new HADDOCK scoring function for protein-peptide docking](https://journal.frontiersin.org/article/10.3389/fmolb.2016.00046/full).
_Frontiers in Molecular Biosciences_, *3*:46 doi:10.3389/fmolb.2016.00046 (2016).

* L.C. Xue, J.P.G.L.M. Rodrigues, P.L. Kastritis, **A.M.J.J. Bonvin**# and A. Vangone#.
[PRODIGY: a web server for predicting the binding affinity of protein-protein complexes](https://doi.org/10.1093/bioinformatics/btw514).
_Bioinformatics_, *32*, 3676-3678 (2016).

* R. Melo, R. Fieldhouse, A. Melo, J.D.G. Correia, M.D.S. Cordeiro, Z.H. Gumus, J. Costa,  **A.M.J.J. Bonvin** and I.S. Moreira.
[A Machine-Learning Approach for Hot-Spot Detection at Protein-Protein Interfaces](https://doi.org/10.3390/ijms17081215).
_International Journal of Molecular Sciences_, *17*, 1215 (2016).

* G.C.P van Zundert and  **A.M.J.J. Bonvin**.
[Defining the limits and reliability of rigid-body fitting in cryo-EM maps using multi-scale image pyramids](https://doi.org/10.1016/j.jsb.2016.06.011).
_J. Struct. Biol._, *195*, 252-258 (2016).

* Y. Liu, J.P.G.L.M. Rodrigues, **A.M.J.J. Bonvin**, E.A. Zaal, C.R. Berkers, M. Heger, K. Gawarecka, E. Swiezewska, E. Breukink and M.R. Egmond.
[New Insight into the Catalytic Mechanism of Bacterial MraY from Enzyme Kinetics and Docking Studies](https://doi.org/10.1074/jbc.M116.717884).
_J. Biol. Chem._, *291*, 15057-15068 (2016).

* C. Geng, A. Vangone and **A.M.J.J. Bonvin**.
[Exploring the interplay between experimental methods and the performance of predictors of binding affinity change upon mutations in protein complexes](https://doi.org/10.1093/protein/gzw020).
_Protein Engineering, Design, and Selection_, *29*, 291-299 (2016).

* L.C. Xue, J.P.G.L.M. Rodrigues, D. Dobbs, V. Honavarand  **A.M.J.J. Bonvin**.
[Template-Based Protein-Protein Docking Exploiting Pairwise Interfacial Residue Restraints](https://doi.org/10.1093/bib/bbw027).
_Briefings in Bioinformatics_, *18*, 458-466 (2016).

* M.F. Lensink, S. Velankar, A. Kryshtafovych, S. Huang, D. Schneidman-Duhovny, A. Sali, J. Segura, N.s Fernandez-Fuentes, S. Viswanath, R. Elber, S. Grudinin, P. Popov, E. Neveu, Hasup Lee, M. Baek, S. Park, L. Heo, G. Rie Lee, C. Seok, S. Qin, H. Zhou, D.W. Ritchie, B. Maigret, M. Devignes, A. Ghoorah, M. Torchala, R.A.G. Chaleil, P.A. Bates, E.t Ben-Zeev, M. Eisenstein, S.S. Negi, Z. Weng, T. Vreven, B.G. Pierce, T. M. Borrman, J. Yu, F. Ochsenbein, R. Guerois, A. Vangone, J.P.G.L.M. Rodrigues, G. van Zundert, M. Nellen, L. Xue, E. Karaca, A.S.J. Melquiond, K. Visscher, P.L. Kastritis, **A.M.J.J. Bonvin**, X. Xu, L. Qiu, C. Yan, J. Li, Z. Ma, J. Cheng, X. Zou, Y. Shen, L.X. Peterson, H. Kim, A. Roy, X. Han, J. Esquivel-Rodriguez, D. Kihara, X. Yu, N.J. Bruce, J.C. Fuller, R.C. Wade, I. Anishchenko, P.J. Kundrotas, I.A. Vakser, K. Imai, K.i Yamada, T. Oda, T. Nakamura, K. Tomii, C. Pallara, M. Romero-Durana, B. Jimenez-Garcia, I.H. Moal, J Fernandez-Recio, J. Young Joung, J. Yun Kim, K.g Joo, J. Lee, D. Kozakov, S. Vajda, S. Mottarella, D.R. Hall, D. Beglov, A. Mamonov, B. Xia, T. Bohnuud, C.A. Del Carpio, E. Ichiishi, N. Marze, D. Kuroda, S.S. Roy Burman, J.J. Gray, E. Chermak, L. Cavallo, R. Oliva, A. Tovchigrechko and S.J. Wodak.
[Prediction of homo- and hetero-protein complexes by ab-initio and template-based docking: a CASP-CAPRI experiment](https://doi.org/10.1002/prot.25007).
_Proteins: Struc. Funct. &amp; Bioinformatics_, *84*, 323-348 (2016)

* P. Sliz, P. Meyer, S. Socias, J. Key, E. Ransey, E.C. Tjon, A. Buschiazzo, M. Lei, C. Botka, J. Withrow, D.B. Neau, K. Rajashankar, K.S. Anderson, R. Baxter, S. Blacklow, T. Boggon, **A.M.J.J. Bonvin**, D. Borek, T. Brett, A. Caflisch, C. Chang, W. Chazin, K.D. Corbett, M. Cosgrove, S. Crosson, S. Dhe-Paganon, E. Di Cera, C. Drennan, M. Eck, B. Eichman, Q. Fan, A. Ferra-D Amara, J.S. Fraser, J.C. Fromme, K. Garcia, R. Gaudet, P. Gong, S. Harrison, E. Heldwein, Z. Jia, R. Keenan, A. Kruse, M. Kvansakul, J. McLellan, Y. Modis, Y. Nam, Zb. Otwinowski, E.F. Pai, P.J. Barbosa Pereira, C. Petosa, C.S. Raman, T. Rapoport, A. Roll-Mecak, M. Rosen, G. Rudenko, J. Schlessinger, T. Schwartz, Y. Shamoo, H. Sondermann, Y. Tao, N. Tolia, O.V. Tsodikov, K.D. Westover, H. Wu, I. Foster, F. Maia, T. Gonen, T. Kirchhausen, K. Diederichs and M. Crosas.
[Data publication with the structural biology data grid supports live analysis](https://doi.org/10.1038/ncomms10882).
_Nature Communications_, *7*: 10882, 1-12 (2016).

* S. Jung, J. Fischer, B. Spudy, K. Tim, F.D. Soennichsen, L. Xue, **A.M.J.J. Bonvin**, P. Goettig, V. Magdolen, U. Meyer-Hoffert, J. Grötzinger.
[The solution structure of the kallikrein-related peptidases inhibitor SPINK6](https://doi.org/10.1016/j.bbrc.2016.01.172).
_Biochemical and Biophysical Research Communications_, *471*, 103-108 (2016).

* E. Deplazes, J. Davies, **A.M.J.J. Bonvin**, G.F. King and A.E. Mark.
[On the Combination of Ambiguous and Unambiguous Data in the Restraint-driven Docking of Flexible Peptides with HADDOCK: The Binding of the Spider Toxin PcTx1 to the Acid Sensing Ion channel (ASIC)1a](https://doi.org/doi:10.1021/acs.jcim.5b00529).
_J. Chem. Inf. and Model._ *56*, 127-138 (2016).

* J.P.G.L.M. Rodrigues, A.S.J. Melquiond and  **A.M.J.J. Bonvin**.
[Molecular Dynamics Characterization of the Conformational Landscape of Small Peptides: A series of hands-on collaborative practical sessions for undergraduate students](https://doi.org/doi:10.1002/bmb.20941).
_Biochemistry and Molecular Biology Education_, *44*, 160-167 (2016).

* R. Zhang, T. Eckert, T. Lutteke, S. Hanstein, A. Scheidig, **A.M.J.J. Bonvin**, N.E. Nifantiev, T. Kozar, R. Schauer, M. Abdulaziz Enani and H.C. Siebert.
[Structure-function relationship of antimicrobial peptides and proteins in respect to contact-molecules on pathogen surfaces](https://europepmc.org/abstract/med/26139116).
_Current Topics in Medicinal Chemistry_. *16*, 89-98 (2016).

* G.C.P van Zundert, J.P.G.L.M. Rodrigues, M. Trellet, C. Schmitz, P.L. Kastritis, E. Karaca, A.S.J. Melquiond, M. van Dijk, S.J. de Vries and  **A.M.J.J. Bonvin**.
[The HADDOCK2.2 webserver: User-friendly integrative modeling of biomolecular complexes](https://doi.org/doi:10.1016/j.jmb.2015.09.014).
_J. Mol. Biol._, *428*, 720-725 (2016).


## 2015
<hr />
* M. Kalia, S. Willkomm, J.C. Claussen, T. Restle# and **A.M.J.J. Bonvin#**.
[Novel insights into guide RNA 5’-nucleoside/tide binding by human Argonaute 2](https://doi.org/doi:10.3390/ijms17010022).
_Int. J. Mol. Sci._ *22*, doi:10.3390/ijms17010022 (2015).

* **A.M.J.J. Bonvin** and O. Keskin. 
[Editorial overview: Protein-Protein Interactions.](https://doi.org/10.1016/j.sbi.2015.11.004)
_Curr. Opin. Struct. Biol._ *35*, vii-ix (2015). ([free PDF download](https://authors.elsevier.com/a/1SBy1,LqAqxhW8))

* L.C. Xue, D. Dobbs. **A.M.J.J. Bonvin** and V. Honavar
[Computational Prediction of Protein Interfaces: A Review of Data Driven Methods](https://doi.org/10.1016/j.febslet.2015.10.003).
_FEBS Letters_, *589*, 3516-3526 (2015).

* F. Marino, M. Bern, G.P.M. Mommen, A.C. Leney, J.A.M. van Gaans-van den Brink, **A.M.J.J. Bonvin**, C. Becker, C.A.C.M. van Els and A.J.R. Heck.
[Extended O-GlcNAc on HLA Class-I-Bound Peptides](https://pubs.acs.org/doi/abs/10.1021/jacs.5b06586).
_J. Am. Chem. Soc._ *137*, 10922-10925 (2015).

* T Vreven+, I.H. Moal+, A. Vangone+, B.G. Pierce, P.L. Kastritis, M. Torchala, R. Chaleil, B. Jiménez-García, P.A. Bates#, Juan Fernandez-Recio#, **A.M.J.J. Bonvin**# and Z. Weng#.
[Updates to the integrated protein-protein interaction benchmarks: Docking benchmark version 5 and affinity benchmark version 2](https://doi.org/doi:10.1016/j.jmb.2015.07.016).
_J. Mol. Biol._ *19*, 3031-3041 (2015).

* A Vangone and **A.M.J.J. Bonvin**.
[Contacts-based prediction of binding affinity in protein-protein complexes](https://elifesciences.org/content/4/e07454).
_eLife_, *4*, e07454 (2015).

* M. Rad-Malekshahi, K. Visscher, J.P.G.L.M. Rodrigues, R. De Vries, W. Hennink, M. Baldus, **A.M.J.J. Bonvin**#, E. Mastrobattista# and M. Weingarth#.
[The supramolecular organization of a peptide-based nanocarrier at high molecular detail](https://pubs.acs.org/doi/abs/10.1021/jacs.5b02919).
_J. Am. Chem. Soc._ *137*, 7775-7784 (2015).

* G.C.P. van Zundert and **A.M.J.J. Bonvin**. 
[DisVis: Quantifying and visualizing accessible interaction space of distance-restrained biomolecular complexes](https://doi.org/doi:10.1093/bioinformatics/btv333).
_Bioinformatics_. *31*, 3222-3224 (2015).

* F. Psomopoulos, Af.M. Duarte, C. Blanchet, **A.M.J.J. Bonvin**, M. Corpas, A. Franc, R.C. Jimenez, J. Marco de Lucas, T. Nyrönen, G. Sipos and S.B. Suhr.
[Future opportunities and future trends for e-infrastructures and life sciences: going beyond grid to enable life science data analysis](https://journal.frontiersin.org/article/10.3389/fgene.2015.00197).
_Frontiers in Genetics_. *6:197*, doi:10.3389/fgene.2015.00197 (2015).

* A. Sali, H.M. Berman, T. Schwede, J. Trewhella, G. Kleywegt, S.K. Burley, J. Markley, H. Nakamura, P. Adams, **A.M.J.J. Bonvin**, W. Chiu, M. Dal Peraro, F. Di Maio, T.E. Ferrin, K. Grünewald, A. Gutmanas, R. Henderson, G. Hummer, K. Iwasaki, G. Johnson, C.L. Lawson, J. Meiler, M.A. Marti-Renom, G.T. Montelione, M. Nilges, R. Nussinov, A. Patwardhan, J. Rappsilber, R.J. Read, H. Saibil, G.F. Schröder, C. Schwieters, C.A.M. Seidel, D. Svergun, M. Topf, E.L. Ulrich, S. Velankar, and J.D. Westbrook.
[Outcome of the First wwPDB Hybrid / Integrative Methods Task Force Workshop](https://doi.org/10.1016/j.str.2015.05.013).
_Structure_. *23*, 1056-1067 (2015).

* G. van der Schot and **A.M.J.J. Bonvin**. 
[Performance of the WeNMR CS-Rosetta3 web server in CASD-NMR](https://doi.org/doi:10.1007/s10858-015-9942-7).
_J. Biomol. NMR_. *62*, 497-502 (2015).

* T. Sinnige, M. Weingarth, M. Daniëls, R. Boelens, **A.M.J.J. Bonvin**, K. Houben and M. Baldus.
[Conformational plasticity of the POTRA 5 domain in the outer membrane protein assembly factor BamA](https://doi.org/doi:10.1016/j.str.2015.04.014).
_Structure_. *23*, 1317-1324 (2015).

* G.C.P. van Zundert and **A.M.J.J. Bonvin**.
[Fast and sensitive rigid-body fitting into cryo-EM density maps with PowerFit](https://doi.org/doi:10.3934/biophy.2015.2.73).
_AIMS Biophysics_. *2*, 73-87 (2015).

* K.D. Brewer, T. Bacaj, A. Cavalli, C. Camilloni, J.D. Swarbrick, J. Liu, A. Zhou, P. Zhou, N. Barlow, J. Xu, A.B. Seven, E.A. Prinslow, R. Voleti, D. Haussinger, **A.M.J.J. Bonvin**, D.R. Tomchick, M. Vendruscolo, B. Graham, T.C. Sudhof, J. Rizo.
[Dynamic Synaptotagmin-1-SNARE Complex Binding Mode in Solution](https://doi.org/doi:10.1038/nsmb.3035).
_Nature Struct. &amp;  Mol. Biol._ *22*, 555-564 (2015).

* G.C.P. van Zundert, A.S.J. Melquiond and **A.M.J.J. Bonvin**.
[Integrative modeling of biomolecular complexes: HADDOCKing with Cryo-EM data.](https://doi.org/10.1016/j.str.2015.03.014)
_Structure._ *23*, 949-960 (2015).

* M. Kaplan, A. Cukkemane, G.C.P. van Zundert, S. Narasimhan, G. Waksman, **A.M.J.J. Bonvin**, R. Fronzes, G.E. Folkers and M. Baldus.
[Probing a cell-embedded Megadalton protein complex by DNP-supported solid-state NMR](https://doi.org/doi:10.1038/nmeth.3406).
_Nature Methods._ *12*, 649-652 (2015).

* K.M. Visscher, P.L. Kastritis and **A.M.J.J. Bonvin**.
[Non-interacting surface solvation and dynamics in protein-protein interactions.](https://doi.org/doi:10.1002/prot.24741)
_Proteins: Struc. Funct. &amp; Bioinformatics_ *83*, 445-458 (2015).

* M. Trellet, A.S.J. Melquiond and **A.M.J.J. Bonvin**.
[Information-driven modelling of protein-peptide complexes.](https://link.springer.com/protocol/10.1007/978-1-4939-2285-7_10)
_Methods in Molecular Biology._ Ed. Peng Zhou. Humana Press Inc. 221-239 (2015)

* J.P.G.L.M Rodrigues, E. Karaca and **A.M.J.J. Bonvin**.
[Information-driven structural modelling of protein-protein interactions.](https://link.springer.com/protocol/10.1007/978-1-4939-1465-4_18)
_Methods in Molecular Biology: Molecular Modelling of Proteins._ Ed. Andreas Kokul. Humana Press Inc. 399-424 (2015).

* W.F. Vranken, G.W. Vuister and **A.M.J.J. Bonvin**.
[NMR-based modelling and refinement of protein 3D structures.](https://link.springer.com/protocol/10.1007/978-1-4939-1465-4_16)
_Methods in Molecular Biology: Molecular Modelling of Proteins._ Ed. Andreas Kokul. Humana Press Inc. 351-380 (2015).


## 2014
<hr />
* F.M. Ferguson, D.M. Dias, J.P.G.L.M. Rodrigues, J. Wienk, R. Boelens, **A.M.J.J. Bonvin**, C. Abell, A. Cuilli.
[Binding Hotspots of BAZ2B Bromodomain: Histone Interaction Revealed by Solution NMR Driven Docking.](https://doi.org/doi:10.1021/bi500909d)
_Biochemistry._ *53*, 6706-6716 (2014).

* T.A. Hopf#, C.P.I. Sch&auml;rfe#, J.P.G.L.M. Rodrigues#, A.G. Green, O. Kohlbacher, C. Sander\*, **A.M.J.J. Bonvin\*** and D.S. Marks\*.
[Sequence co-evolution gives 3D contacts and structures of protein complexes.](https://elifesciences.org/content/3/e03430)
_eLife._ *3* e03430 (2014)

* M. Rey, V. Sarpe, K. Burns, J. Buse, Ch.A.H. Baker, M. van Dijk, L. Wordeman, **A.M.J.J. Bonvin** and D. Schriemer.
[Mass Spec Studio for Integrative Structural Biology.](https://doi.org/10.1016/j.str.2014.08.013)
_Structure._ *22*, 1538-1548 (2014).

* P.L. Kastritis, J.P.G.L.M Rodrigues, G.E. Folkers, R. Boelens and **A.M.J.J. Bonvin**.
[Proteins feel more than they see: Fine-tuning of binding affinity by properties of the non-interacting surface.](https://doi.org/10.1016/j.jmb.2014.04.017)
_J. Mol. Biol._ *426*, 2632-2652 (2014). <a href="https://f1000.com/prime/718363743?bd=1&amp;ui=155629" target="_blank"><img src="https://cdn.f1000.com.s3.amazonaws.com/images/badges/badgef1000.gif" alt="Access the recommendation on F1000Prime" id="bg" name="bg"></a>

* J.P.G.L.M Rodrigues and **A.M.J.J. Bonvin**
[Integrative computational modeling of protein interactions.](https://doi.org/doi:10.1111/febs.12771)
_FEBS J._, *281*, 1988-2003 (2014).

* P.L. Kastritis, J.P.G.L.M. Rodrigues and **A.M.J.J. Bonvin**
[HADDOCK2P2I: A robust biophysical model for predicting the binding affinity of protein-protein interaction inhibitors](https://pubs.acs.org/doi/abs/10.1021/ci4005332)
_J. Chem. Info. Model._ *54*, 826-836 (2014).

* H. van Ingen and **A.M.J.J. Bonvin**
[Information-driven modeling of large macromolecular assemblies using NMR data.](https://doi.org/doi:10.1016/j.jmr.2013.10.021)
_J. Magn. Reson._ *241*, 103-114 (2014).

* J. Snijder, R.J. Burnley, A. Wiegard, A.S.J. Melquiond, **A.M.J.J. Bonvin**, I.M. Axmann, A.J.R. Heck
[Insight into cyanobacterial circadian timing from structural details of the KaiB-KaiC interaction.](https://www.pnas.org/content/111/4/1379.short)
_Proc. Natl. Acad. Sci. USA._ *111*, 1379-1383 (2014).

* G.C.P. van Zundert and **A.M.J.J. Bonvin**.
[Modeling protein-protein complexes using the HADDOCK webserver.](https://link.springer.com/protocol/10.1007/978-1-4939-0366-5_12)
_Methods in Molecular Biology: Protein Structure Prediction._ Ed. Daisuke Kihara. Humana Press Inc., 163-179 (2014).

* M.F. Lensink, I.H. Moal, P.A. Bates, P.L. Kastritis, A.S.J. Melquiond, E. Karaca, C. Schmitz, M. van Dijk, **A.M.J.J. Bonvin**, M. Eisenstein, B. Jimenez-Garcia, S. Grosdidier, S.A Solernou, L. Perez-Cano, C. Pallara, J. Fernandez-Recio, J. Xu, P. Muthu, K. Praneeth Kilambi, J.J. Gray, S. Grudinin, G. Derevyanko, J.C. Mitchell, J. Wieting, E. Kanamori, Y. Tsuchiya, Y. Murakami, J. Sarmiento, D.M. Standley, M. Shirota, K. Kinoshita, H. Nakamura M. Chavent, D.W. Ritchie, H. Park, J. Ko, H. Lee, C. Seok, Y. Shen, D. Kozakov, S. Vajda, P. J. Kundrotas, I.A. Vakser, B.G. Pierce, H. Hwang, T. Vreven, Z. Weng, I. Buch, E. Farkash, H.J. Wolfson, M. Zacharias, S. Qin, H.-X. Zhou, S.-Y. Huang, X. Zou, J.A. Wojdyla, C. Kleanthous and S.J. Wodak.
[Blind Prediction of Interfacial Water Positions in CAPRI.](https://doi.org/doi:10.1002/prot.24439)
_Proteins: Struc. Funct. &amp; Bioinformatics_ *82*, 620-632 (2014).


## 2013
<hr />

* K.M. Varney#, **A.M.J.J. Bonvin**#, M. Pazgier#, J. Malin#, W. Yu#, E. Ateh, T. Oashi, J. Huang, W. Lu, M. Diepeveen-de Buin, J. Bryant, E. Breukink, A.D. MacKerell Jr. and E.P.H. de Leeuw
[Turning Defense into Offense: Development of Defensin Mimetics as Novel Antibiotics targeting Lipid II.](https://www.plospathogens.org/article/info:doi/10.1371/journal.ppat.1003732)
_PloS Pathogens_, *9*, e1003732. doi:10.1371/journal.ppat.1003732 (2013).

* G. van der Schot, Z. Zhang, R. Vernon, Y. Shen, W.F. Vranken, D. Baker, **A.M.J.J. Bonvin\*** and O.F. Lange\*
[Improving 3D structure prediction from chemical shift data.](https://doi.org/doi:10.1007/s10858-013-9762-6)
_J. Biomol. NMR._, *57*, 27-35 (2013).

* **A.M.J.J. Bonvin**
[Coming to peace with protein complexes? 5th CAPRI evaluation meeting, April 17-19th 2013, Utrecht.](https://doi.org/doi:10.1002/prot.24431)
_Proteins: Struc. Funct. &amp; Bioinformatics_ Editorial, CAPRI special issue. *81*, 20173-2074 (2013).

* J.P.G.L.M. Rodrigues, A.S.J. Melquiond, E. Karaca, M. Trellet, M. Van Dijk, G.C.P. Van Zundert, C. Schmitz, S.J. de Vries, A. Bordogna, L. Bonati, P.L. Kastritis and **A.M.J.J. Bonvin**
[Defining the limits of homology modelling in information-driven protein docking](https://doi.org/doi:10.1002/prot.24382)
_Proteins: Struc. Funct. &amp; Bioinformatics_, *81*, 2119-2128 (2013).

* R. Morretti et al.
[Community-wide Evaluation of Methods for Predicting the Effect of Mutations on Protein-Protein Interactions.](https://doi.org/doi:10.1002/prot.24356)
_Proteins: Struc. Funct. &amp; Bioinformatics_, *81*, 1980-1987 (2013).

* J. Janin and **A.M.J.J. Bonvin**
[Protein-Protein Interactions.](https://doi.org/10.1016/j.sbi.2013.10.003)
_Curr. Opin. Struct. Biol._ Editorial, special issue on protein-protein interactions. *23*, 859-861 (2013).

* P.L. Kastritis and **A.M.J.J. Bonvin**
[Molecular origins of binding affinity: Seeking the Archimedean point.](https://doi.org/doi:10.1016/j.sbi.2013.07.001)
_Curr. Opin. Struct. Biol._, *23*, 868-877 (2013).

* D.M. Dias, J.P.G.L.M. Rodrigues, N.S. Domingues, **A.M.J.J. Bonvin** and M.C.A. Castro.
[Unveiling the Interaction of Vanadium Compounds with Human Serum Albumin using 1H STD NMR and Computational Docking Studies.](https://doi.org/doi:10.1002/ejic.201300419)
_Eur. J. Inorg. Chem._, *26*, 4619-4626 (2013).

* E.A.W. van der Cruijsen, D. Nand, M. Weingarth, A, Prokofyev, S, Hornig, **A.M.J.J. Bonvin**, S. Becker, R.E. Hulse, E. Perozo, O. Pongs, M. Baldus.
[The importance of the lipid-pore loop interface for potassium channel structure and function.](https://doi.org/doi:10.1073/pnas.1305563110)
_Proc Natl. Acad. Sci. USA_, *110*, 13008-13013. (2013).

* M. van Dijk, K. Visscher, P.L. Kastritis and **A.M.J.J. Bonvin**.
[Solvated protein-DNA docking using HADDOCK.](https://doi.org/doi:10.1007/s10858-013-9734-x)
_J. Biomol. NMR_, *56*, 51-63 (2013).

* E. Karaca and **A.M.J.J. Bonvin**.
[On the usefulness of Ion Mobility Mass Spectrometry and SAXS data in scoring docking decoys.](https://doi.org/doi:10.1107/S0907444913007063)
_Acta Cryst. D._, *D69*, 683-694 (2013).

* M. Weingarth, A. Prokofyev, E. van der Cruijsen, D. Nand, **A.M.J.J. Bonvin**, O. Pongs and M. Baldus
[Structural determinants and functional aspects of specific lipid binding to potassium channels.](https://doi.org/doi:10.1021/ja3119114)
_J. Am. Chem. Soc._, *35*, 3983-3988 (2013).

* S. Alex, K. Lange, T. Amolo, J.S. Grinstead, E. Szalowska, A. Koppen, K. Mudde, D. Haenen, S. Al-Lahham, H. Roelofsen, R. Houtman, B. van der Burg, **A.M.J.J. Bonvin**, E. Kalkhoven, M. Muller, G. Hooiveld and S. Kersten.
[Short chain fatty acids stimulate Angptl4 synthesis in human colonocytes by selective PPAR modulation.](https://doi.org/doi:10.1128/MCB.00858-12)
_Mol. Cell. Biol._, *33*, 1303-1316 (2013).

* D.L. Hulsik, Y.Y. Liu, N.M. Strokappe, S. Battella, M. El Khattabi, L.E. McCoy, A. Hinz, M. Hock, **A.M.J.J. Bonvin**, J.P.M. Langedijk, D. Davis, A. Forsman Quinley, M. Aasa-Chapman, R.A. Weiss, C.T. Verrips, W. Weissenhorn and L. Rutten
[A gp41 MPER-specific Llama VHH Requires a Hydrophobic CDR3 for Neutralization but not for Antigen Recognition.](https://doi.org/doi:10.1371/journal.ppat.1003202)
_PloS Pathogens_, *9*, e1003202 (2013).

* M. Trellet, A.S.J. Melquiond and **A.M.J.J. Bonvin**.
[A Unified Conformational Selection and Induced Fit Approach to Protein-Peptide Docking](https://dx.plos.org/10.1371/journal.pone.0058769)
_PLoS ONE_, *8(3)* e58769 (2013). --> [**ERRATUM**](https://milou.science.uu.nl/Files/ERRATUM_unified_conformational_selection.pdf)

* R. Dagil, C O'Shea, A. Nykaer, **A.M.J.J. Bonvin** and B.B. Kragelund
[Gentamicin Binds to Megalin as a Competitive Inhibitor and Utilizes the Common Ligand Binding Motif of Complement Type Repeats.](https://doi.org/doi:10.1074/jbc.M112.434159)
_J. Biol. Chem_, *288*, 4424-4435 (2013).

* E. Karaca and **A.M.J.J. Bonvin**
[Advances in integrated modelling of biomolecular complexes.](https://doi.org/doi:10.1016/j.ymeth.2012.12.004)
_Methods_, *59*, 372-381 (2013).

* P.L. Kastritis and **A.M.J.J. Bonvin**
[On the binding affinity of macromolecular interactions: daring to ask why proteins interact](https://doi.org/doi:10.1098/rsif.2012.0835)
_J. R. Soc. Interface_, *10*, doi: 10.1098/rsif.2012.0835 (2013).

* **A.M.J.J. Bonvin**, M. van Dijk, E. Karaca, P.L. Kastritis, A.S.J. Melquiond, C. Schmitz and S.J. de Vries
[HADDOCK](https://www.springer.com/life+sciences/biochemistry+%26+biophysics/book/978-3-642-16711-9)
In Encyclopedia of Biophysics, Ed. G.C.K. Roberts, Springer-Verlag Berlin Heidelberg (2013).

* P.L. Kastritis, K.M. Visscher, A.D.J. van Dijk and **A.M.J.J. Bonvin**
[Solvated docking using Kyte-Doolittle-based water propensities.](https://doi.org/doi:10.1002/prot.24210)
_Proteins: Struc. Funct. &amp; Bioinformatic._, *81*, 510-518 (2013).


## 2012
<hr />

* S. Gradmann, C. Ader, I. Heinrich, D. Nand, M. Dittmann, A. Cukkemane, M. van Dijk, **A.M.J.J. Bonvin**, M. Engelhard and M. Baldus
[Rapid Prediction of multi-dimensional NMR data sets.](https://doi.org/doi:10.1007/s10858-012-9681-y)
_J. Biomol. NMR_, *54*, 377-387 (2012).

* J. Hennig, S. de Vries, K.D.M. Hennig, L. Randles, K.J. Walters, M. Sunnerhagen and **A.M.J.J. Bonvin**
[MTMDAT-HADDOCK: High-throughput, protein complex structure modeling based on limited proteolysis and mass spectrometry.](https://doi.org/doi:10.1186/1472-6807-12-29)
_BMC Struc. Biol._, *12*, 29 (2012).

* S.J.L. van Wijk, A.S.J. Melquiond, S.J. de Vries, H.Th.M Timmers and **A.M.J.J. Bonvin**
[Dynamic control of selectivity in E2-E3 interactions via a salt-bridge network revealed by an ASP to GLU substitution.](https://www.ploscompbiol.org/article/info:doi/10.1371/journal.pcbi.1002754)
_PLoS Comp. Biol._, *8(11)*, e1002754 (2012).

* T.A. Wassenaar, M. van Dijk, N. Loureiro-Ferreira, G. van der Schot, S.J. de Vries, C. Schmitz, J. van der Zwan, R. Boelens, A. Giachetti, L. Ferella, A. Rosato, I. Bertini, T. Herrmann, H.R.A. Jonker, A. Bagaria, V. Jaravine, P. Guntert, H. Schwalbe, W.F. Vranken, J.F. Doreleijers, G. Vriend, G.W. Vuister, D. Franke, A. Kikhney, D.I. Svergun, R. Fogh, J. Ionides, E.D. Laue, C. Spronk, S. Jurka, M. Verlato, S. Badoer, S. Dal Pra, M. Mazzucato, E. Frizziero and **A.M.J.J. Bonvin**
[WeNMR: Structural Biology on the Grid.](https://link.springer.com/article/10.1007%2Fs10723-012-9246-z)
_J. Grid. Comp._, *10*, 743-767 (2012).

* T.A. Wassenaar, S.J.L. de Vries, H. Bekker and **A.M.J.J. Bonvin**
[SQUEEZE-E: The optimal solution for molecular simulations with periodic boundary conditions.](https://doi.org/doi:10.1021/ct3000662)
_J. Chem. Theo. Comput._, *8*, 3618-3627 (2012).

* M. van Dijk, T.A. Wassenaar and **A.M.J.J. Bonvin**
[A flexible, grid-enabled web portal for GROMACS molecular dynamics simulations](https://doi.org/doi:10.1021/ct300102d)
_J. Chem. Theo. Comput._, *8*, 3463-3472 (2012).

* P.L. Kastritis and **A.M.J.J. Bonvin**
[Predicting and Dissecting High-order Molecular Complexity by Information-driven Biomolecular Docking](https://bookshop.cabi.org/?site=191&amp;page=2633&amp;pid=2443)
In: <i>Antimicrobial Drug Discovery. Emerging Strategies</i> Adv. Mol. Cell. Microbiol., Eds G. Tegos and E. Mylokanis, CABI, *22*, 232-246 (2012).

* M. Weingarth, C. Ader, A.S.J. Melquiond, D. Nand, O. Pongs, S. Becker, **A.M.J.J. Bonvin** and M. Baldus.
[Supramolecular structure of membrane-associated polypeptides by combining solid-state NMR and Molecular Dynamics simulations.](https://doi.org/doi:10.1016/j.bpj.2012.05.016)
_Biophys. J._, *103*, 29-37 (2012).

* J.P.G.L.M. Rodrigues, M. Trellet, C. Schmitz, P.L. Kastritis, E. Karaca, A.S.J. Melquiond and **A.M.J.J. Bonvin**
[Clustering biomolecular complexes by residue contacts similarity.](https://doi.org/doi:10.1002/prot.24078)
_Proteins: Struc. Funct. &amp; Bioinformatic_, *80*, 1810-1817 (2012).

* C. Schmitz, A.S.J. Melquiond, S.J. de Vries, E. Karaca, M. van Dijk, P.L. Kastritis and **A.M.J.J. Bonvin**
Protein-protein docking with HADDOCK
In: <a href="https://eu.wiley.com/WileyCDA/WileyTitle/productCd-3527328505,subjectCd-CH17.html">NMR in Mechanistic Systems Biology</a>. Ed. I. Bertini, K.S. McGreevy and G. Parigi, Wiley-Blackwell, 512-535.

* A. Rosato, J. M. Aramini, C. Arrowsmith, A. Bagaria, D. Baker, A. Cavalli, J.F. Doreleijers, A. Eletsky, A. Giachetti, P. Guerry, A. Gutmanas, P. Guntert, Y. He, T. Herrmann, Y.J. Huang, V. Jaravine, H.R.A. Jonker, M.A. Kennedy, O.F. Lange, G. Liu, T.E. Malliavin, R. Mani, B. Mao, G.T. Montelione, M. Nilges, P. Rossi, G. van der Schot, H. Schwalbe, T.A. Szyperski, M. Vendruscolo, R. Vernon, W.F. Vranken, S.J. de Vries, G.W. Vuister, B. Wu, Y. Yang and **A.M.J.J. Bonvin**
[Blind testing of routine, fully automated determination of protein structures from NMR data.](https://doi.org/doi:10.1016/j.str.2012.01.002)
_Structure_, *20*, 227-236 (2012).

* P.L. Kastritis, A.D.J. van Dijk and **A.M.J.J. Bonvin**
[Explicit Treatment of Water Molecules in Data-Driven Protein-Protein Docking: The Solvated HADDOCKing Approach](https://doi.org/doi:10.1007/978-1-61779-465-0_22)
_Methods in Molecular Biology_ *819*, Part 5, 355-374 (2012)

* A.S.J. Melquiond, E. Karaca, P.L. Kastritis and **A.M.J.J. Bonvin**
[Next challenges in protein-protein docking: From proteome to interactome and beyond.](https://onlinelibrary.wiley.com/doi/10.1002/wcms.91/full)
_WIREs Computational Molecular Science_, *2*, 642-651 (2012).

* T.A. Wassenaar, M. van Dijk, N. Loureiro-Ferreira, G. van der Schot, S.J. de Vries, C. Schmitz, J. van der Zwan, R. Boelens, A. Giachetti, L. Ferella, A. Rosato, I. Bertini, T. Herrmann, H.R.A. Jonker, A. Bagaria, V. Jaravine, P. Guntert, H. Schwalbe, W.F. Vranken, J.F. Doreleijers, G. Vriend, G.W. Vuister, D. Franke, A. Kikhney, D.I. Svergun, R. Fogh, J. Ionides, E.D. Laue, C. Spronk, M. Verlato, S. Badoer, S. Dal Pra, M. Mazzucato, E. Frizziero and **A.M.J.J. Bonvin**
[WeNMR: Structural Biology on the Grid.](https://ceur-ws.org/Vol-819/)
In: <i>IWSG-Life 2011: Science Gateways for Life Sciences 2011</i>. Proceedings of the 3rd International Workshop on Science Gateways for Life Sciences London, United Kingdom. June 8-10, 2011. Eds G. Terstyanszky and T. Kiss. urn:nbn:de:0074-819-5


## 2011
<hr />

* S.J Fleishman, <i>et al.</i>
[Community-wide assessment of protein-interface modeling suggests improvements to design methodology](https://doi.org/10.1016/j.jmb.2011.09.031)
_J. Mol. Biol._, *414*, 289-302 (2011).

* A.M. Wu, T, Singh, J.-H, Liu, S, Andre, M, Lensch, H-C, Siebert, M, Krzeminski, **A.M.J.J. Bonvin**, H, Kaltner, J,H. Wu and H.-J. Gabius
[Adhesion/Growth-Regulatory Galectins: Insights into Their Ligand Selectivity Using Natural Glycoproteins and Glycotopes](https://doi.org/doi:10.1007/978-1-4419-7877-6_7)
In: <i>The Molecular Immunology of Complex Carbohydrates-3</i>, Advances in Experimental Medicine and Biology. A.M. Wu (ed.). Springer Science+Business Media, LLC. pp. 117-141 (2011)

* C. Schmitz and **A.M.J.J. Bonvin**
[Protein-Protein HADDocking using exclusively Pseudocontact Shifts.](https://doi.org/doi:10.1007/s10858-011-9514-4)
_J. Biomol. NMR_, *50*, 263-266 (2011).

* D. Stratman, R. Boelens and **A.M.J.J. Bonvin**
[Quantitative use of chemical shifts for the modelling of protein complexes.](https://doi.org/doi:10.1002/prot.23090)
_Proteins: Struc. Funct. &amp; Bioinformatic_, *79*, 2662-2670 (2011).

* E. Escobar-Cabrera, Okon, D.K.W. Lau, C.F. Dart, **A.M.J.J. Bonvin** and L.P. McIntosh
[Characterizing the N- and C-terminal SUMO interacting motifs of the scaffold protein DAXX.](https://doi.org/doi:10.1074/jbc.M111.231647)
_J. Biol. Chem._, *286*, 19816-19829 (2011).

* Y.C. Fiamegos, P.L. Kastritis, V. Exarchou, H. Han, **A.M.J.J. Bonvin**, J. Vervoort, K. Lewis, M.R. Hamblin and G.P. Tegos
[Antimicrobial and efflux pump inhibitory activity of caffeoylquinic acids from Artemisia absinthium against Gram-positive pathogenic bacteria.](https://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0018127)
_PlosONE_, *6* e18127 (2011).

* S.J. de Vries and **A.M.J.J. Bonvin**
[CPORT: a Consensus Interface Predictor and its Performance in Prediction-driven Docking with HADDOCK.](https://doi.org/doi:10.1371/journal.pone.0017695)
_PlosONE_, *6* e17695 (2011).

* C. Koehler, L. Carlier, D. Veggi, E. Balducci, F. Di Marcello, M. Ferrer-Navarro, M. Pizza, X. Daura, M. Soriani, R. Boelens and **A.M.J.J. Bonvin**
[Structural and biochemical characterization of Nar E, an iron containing ADP-ribosyltransferase from Neisseria meningitides.](https://doi.org/doi:10.1074/jbc.M110.193623)
_J. Biol. Chem._, *286*, 14842-14851 (2011).

* E. Karaca and **A.M.J.J. Bonvin**
[A multi-domain flexible docking approach to deal with large conformational changes in the modeling of biomolecular complexes.](https://doi.org/doi:10.1016/j.str.2011.01.014)
_Structure_, *19* 555-565(2011).

* P.L. Kastritis, I.H. Moal, H. Hwang, Z. Weng, P.A. Bates, **A.M.J.J. Bonvin** and J. Janin
[A structure-based benchmark for protein-protein binding affinity.](https://doi.org/doi:10.1002/pro.580)
_Prot. Sci._, *20*, 482-41 (2011).

* A. Neumoin, A. Leonchiks, P. Petit, L. Vuillard, M. Pizza, M. Soriani, R. Boelens and **A.M.J.J. Bonvin**
[1H, 13C and 15N assignment of the GNA1946 outer membrane lipoprotein from Neisseria meningitidis.](https://doi.org/doi:10.1007/s12104-010-9285-y)
_Biomol. NMR. Assign._, DOI 10.1007/s12104-010-9285-y.

* M. Krzeminski, T. Singh, S. Andre, M. Lensch, A.M. Wu, **A.M.J.J. Bonvin** and H.-J. Gabius
[Human galectin-3 (Mac-2 antigen): defining molecular switches of affinity to natural glycoproteins, structural and dynamic aspects of glycan binding by flexible ligand docking and putative regulatory sequences in the proximal promoter region.](https://doi.org/doi:10.1016/j.bbagen.2010.11.001)
_Biochimica et Biophysica Acta_, *1810*, 150-161 (2011).

* L. Carlier, C. Koehler, D. Veggi, M. Pizza, M. Soriani, R. Boelens and **A.M.J.J. Bonvin**
[NMR resonance assignments of NarE, a putative ADP-ribosylating toxin from Neisseria meningitides.](https://doi.org/doi:10.1007/s12104-010-9261-6)
_Biomol. NMR. Assign._, DOI 10.1007/s12104-010-9261-6.


## 2010
<hr />

* G. Nicastro, S. Todi, E. Karaca, **A.M.J.J. Bonvin**, H. Paulson, A. Pastore
[Understanding the polyUb binding and cleavage properties of ataxin-3.](https://doi.org/doi:10.1371/journal.pone.0012430)
_PlosONE_, *5*, e12430 (2010).

* S.J. de Vries, A.S.J. Melquiond, P.L. Kastritis, E. Karaca, A. Bordogna, M. van Dijk, J.P.G.L.M. Rodrigues and **A.M.J.J. Bonvin**
[Strengths and weaknesses of data-driven docking in CAPRI](https://doi.org/doi:10.1002/prot.22814)
_Proteins: Struc. Funct. &amp; Bioinformatic_, *78*, 3242-3249 (2010).

* T. Schneider, Th. Kruse, R. Wimmer, I. Wiedemann, V. Sass, U. Pag, A. Jansen, A.K. Nielsen, P.H. Mygind, D.S. Raventos, S. Neve, B. Ravn, **A.M.J.J. Bonvin**, L. De Maria, L. Kamenova, H.-G. Sahl and H.-H. Kristensen
[Plectasin, a fungal defensin antibiotic peptide, targets the bacterial cell wall precursor Lipid II.](https://doi.org/doi:10.1126/science.1185723)
_Science_, *328*, 1168-1172 (2010). Download the final author version <a href="https://igitur-archive.library.uu.nl/chem/2011-0314-200259/UUindex.html">here</a>.

* A.S.J. Melquiond and **A.M.J.J. Bonvin**
[Data-driven docking: using external information to spark the biomolecular rendez-vous.](https://www.icpress.co.uk/lifesci/p618.html)
In: Protein-protein complexes: analysis, modelling and drug design. Edited by M. Zacharrias, Imperial College Press, 2010. p 183-209.

* P. Kastritis and **A.M.J.J. Bonvin**
[Are scoring functions in protein-protein docking ready to predict interactomes? Clues from a novel binding affinity benchmark.](https://doi.org/doi:10.1021/pr9009854)
_J. Proteome Research_, *9*, 2216-2225 (2010).  See also the <a href="https://doi.org/doi:10.1021/pr101118t">published correction</a>

* E. Karaca, A.S.J. Melquiond, S.J. de Vries, P.L. Kastritis and **A.M.J.J. Bonvin**
[Building macromolecular assemblies by information-driven docking: Introducing the HADDOCK multi-body docking server.](https://doi.org/doi:10.1074/mcp.M000051-MCP201)
_Mol. Cell. Proteomics_, *9*, 1784-1794 (2010). Download the final author version <a href="https://igitur-archive.library.uu.nl/chem/2011-0314-200254/UUindex.html">here</a>.

* M. van Dijk and **A.M.J.J. Bonvin**
[Pushing the limits of what is achievable in protein-DNA docking. Benchmarking HADDOCK's performance.](https://doi.org/doi:10.1093/nar/gkq222)
_Nucl. Acids Res._, *38*, 5634-5647 (2010).

* **A.M.J.J. Bonvin**, A. Rosato and T.A. Wassenaar
[The eNMR platform for structural biology.](https://doi.org/doi:10.1007/s10969-010-9084-9)
_J. Struct. Funct. Genomics_, *11*, 1-8 (2010).

* S.J. de Vries, M. van Dijk and **A.M.J.J. Bonvin**
[The HADDOCK web server for data-driven biomolecular docking.](https://www.nature.com/nprot/journal/v5/n5/abs/nprot.2010.32.html)
_Nature Protocols_, *5*, 883-897 (2010).

* M. Krzeminski, K. Loth, R. Boelens and **A.M.J.J. Bonvin**
[SAMPLEX: Automatic mapping of perturbed and unperturbed regions of proteins and complexes.](https://www.biomedcentral.com/1471-2105/11/51)
_BMC Bioinformatics_, *11*, 51 (2010).

* M. Kobayashi, E. AB, **A.M.J.J. Bonvin** and G. Siegal
[Structure of the DNA bound BRCT region from human Replication Factor C p140 and model of the protein-DNA complex.](https://doi.org/doi:10.1074/jbc.M109.054106)
_J. Biol. Chem._, *285*, 10087-10097 (2010).


## 2009
<hr />

* A. Rosato, A. Bagaria, Da. Baker, B. Bardiaux, A. Cavalli, J.F. Doreleijers, A. Giachetti, P. Guerry, P. Güntert, T. Herrmann, Y.J. Huang, H.R.A. Jonker, B. Mao, T.E. Malliavin, G.T. Montelione, M. Nilges, S. Raman, G. van der Schot, W.F. Vranken, G.W. Vuister and **A.M.J.J. Bonvin**
[CASD-NMR: Critical Assessment of Automated Structure Determination by NMR.](https://doi.org/doi:10.1038/nmeth0909-625)
_Nature Methods_, *6*, 625-626 (2009).

* S.J. de Vries, M. van Dijk and **A.M.J.J. Bonvin**
[The Prediction of Macromolecular Complexes by Docking.](https://onlinelibrary.wiley.com/doi/10.1002/9780470741894.ch9/summary)
In: Prediction of Protein Structures, Functions, and Interactions (ed J. M. Bujnicki), John Wiley &amp; Sons, Ltd, Chichester, UK (2009).

* D. Eggink, J.P.M. Langedijk, **A.M.J.J. Bonvin**, Y. Deng, M. Lu, B. Berkhout and R.W. Sanders
[Detailed mechanistic insights into HIV-1 sensitivity to three generations fusion inhibitors.](https://doi.org/doi:10.1074/jbc.M109.004416)
_J. Biol. Chem._, *284*, 26941-26950 (2009).

* S.J.L. van Wijk, S.J. de Vries, P. Kemmeren, A. Huang, R. Boelens, **A.M.J.J. Bonvin** and H.Th. M. Timmers
[A comprehensive framework of E2-RING E3 interactions of the human ubiquitin-proteasome system.](https://doi.org/doi:10.1038/msb.2009.55)
_Mol. Sys. Biol._, *5*, 295 (2009).

* J. Romanuka, G.E. Folkers, N. Biris, E. Tishchenko, H. Wienk, **A.M.J.J. Bonvin**, R. Kaptein and R. Boelens
[Specificity and Affinity of Lac Repressor for the Auxiliary Operators O2 and O3 Are Explained by the Structures of Their Protein-DNA Complexes.](https://doi.org/doi:10.1016/j.jmb.2009.05.022)
_J. Mol. Biol._, *390*, 478-489 (2009).

* M. van Dijk and **A.M.J.J. Bonvin**
[3D-DART: a DNA structure modelling server.](https://doi.org/doi:10.1093/nar/gkp287)
_Nucl. Acids Res._, *37*, W235-W239 (2009).

* L. Rutten, J.-P.B.A. Mannie, C.M. Stead, C.R.H. Raetz, C.M. Reynolds, **A.M.J.J. Bonvin**, J. Tommassen, M.R. Egmond, M.S. Trent and Piet Gros
[Active site architecture and catalytic mechanism of the lipid A deacylase LpxR of Salmonella typhimurium.](https://doi.org/doi:10.1073/pnas.0813064106)
_Proc. Natl. Acad. Sci._, *106*, 1960-1964 (2009).

* T. Stockner, S. de Vries, **A.M.J.J. Bonvin**, G. Ecker, P. Chiba
[Data-driven homology modelling of P-glycoprotein in the ATP-bound state indicates flexibility of the transmembrane domains.](https://doi.org/doi:10.1111/j.1742-4658.2008.06832.x)
_FEBS J._, *276*, 964-972 (2009).

* S. Jung, A.J. Dingley, R. Augustin, F. Anton-Erxleben, M. Stanisak, C. Gelhaus, T. Gutsmann, M.U. Hammer, R. Podschun, **A.M.J.J. Bonvin**, M. Leippe, T.C.G. Bosch and J. Groetzinger
[Hydramacin-1: Structure and antibacterial activity of a protein from the basal metazoan Hydra](https://doi.org/doi:10.1074/jbc.M804713200)
_J. Biol. Chem._, *284*, 1896-1905 (2009).

* K. Repanas, G. Fuentes, S.X. Cohen, **A.M.J.J. Bonvin** and Anastassis Perrakis
[Insights into the DNA cleavage mechanism of human LINE-1 retrotransposon endonuclease.](https://doi.org/doi:10.1002/prot.22201)
_Proteins: Struc. Funct. &amp; Bioinformatic_, *74*, 917-928 (2009).

* M. Krzeminski, G. Fuentes, R. Boelens and **A.M.J.J. Bonvin**
[MINOES: a new approach to select a representative ensemble of structures in NMR studies of (partially) unfolded states. Application to d25-PYP.](https://doi.org/doi:10.1002/prot.22197)
_Proteins: Struc. Funct. &amp; Bioinformatic_, *74*, 894-905 (2009).


## 2008
<hr />

* C.J. Arnusch, **A.M.J.J. Bonvin**, A.M. Verel, W.T.M. Jansen, R.M.J. Liskamp, B. de Kruijff, R.J. Pieters, E. Breukink
[Vancomycin-nisin(1-12) conjugate restores activity against vancomycin resistant enterococci (VRE)](https://doi.org/doi:10.1021/bi801597b)
_Biochemistry_, *47*, 12661-12663 (2008).

* R.W. Sanders, S.-T. H. Hsu, E. van Anken, M. Liscaljet, M. Dankers, I. Bontjer, A. Land, I. Braakman, **A.M.J.J. Bonvin** and B. Berkhout
[Evolution Rescues Folding of HIV-1 Envelope Glycoprotein GP120 Lacking a Conserved Disulfide Bond](https://doi.org/doi:10.1091/mbc.E08-07-0670)
_Mol. Biol. Cell_, *19*, 4707-4716 (2008).

* M. van Dijk and **A.M.J.J. Bonvin**
[A protein-DNA docking benchmark.](https://doi.org/doi:10.1093/nar/gkn386)
_Nucl. Acids Res._ (2008), *36*, e88, doi: 10.1093/nar/gkn386.

* S. de Vries and **A.M.J.J. Bonvin**
[How proteins get in touch: Interface prediction in the study of biomolecular complexes.](https://www.bentham-direct.org/pages/content.php?CPPS/2008/00000009/00000004/0005K.SGM)
_Curr. Pept. and Prot. Research_, *9*, 394-406 (2008).

* F. Correa, R. Kopke Salinas, **A.M.J.J. Bonvin** and C.S. Farah
[Deciphering the role of the electrostatic interactions in the a-Tropomyosin head-to-tail complex.](https://doi.org/doi:10.1002/prot.22116)
_Proteins: Struc. Funct. &amp; Bioinformatic_, *73*, 902-917 (2008).

* G. Fuentes, A.D.J. van Dijk and **A.M.J.J. Bonvin**
[NMR-based modelling and refinement of protein 3D structures and their complexes.](https://www.springer.com/humana+press/book/978-1-58829-864-5)
In: _Methods in Molecular Biology 443: Molecular Modelling of Proteins_. A. Kukol Ed., Humana Press, pp. 175-186 (2008).


## 2007
<hr />

* I. Gelis, **A. M. J. J. Bonvin**, D. Keramisanou, M. Koukaki, G. Gouridis, S. Karamanou, A. Economou and C.G. Kalodimos
[Structural basis for signal sequence recognition by the translocase motor SecA determined by NMR](https://www.cell.com/content/article/abstract?uid=PIIS009286740701269X)
_Cell_, *131*, 756-769 (2007).

* S.J. de Vries, A.D.J. van Dijk, M. Krzeminski, M. van Dijk, A. Thureau, V. Hsu, T. Wassenaar and **A. M. J. J. Bonvin**
[HADDOCK versus HADDOCK: New features and performance of HADDOCK2.0 on the CAPRI targets.](https://doi.org/doi:10.1002/prot.21723)
_Proteins: Struc. Funct. &amp; Bioinformatic_ *69*, 726-733 (2007).

* S. Tomaselli, L. Ragona, L. Zetta, M. Assfalg, P. Ferranti, R. Longhi, **A. M. J. J. Bonvin** and H. Molinari.
[NMR-based modeling and binding studies of a ternary complex between chicken liver bile acid binding protein and bile acids.](https://doi.org/doi:10.1002/prot.21517)
_Proteins: Struc. Funct. &amp; Bioinformatic_, *69* 177-191 (2007).

* A.D.J. van Dijk, S. Ciofo-Bafoni, L. Banci, I. Bertini, R. Boelens and **A. M. J. J. Bonvin**.
[Modelling protein-protein complexes involved in the cytochrome c oxidase copper-delivery pathway.](https://doi.org/doi:10.1021/pr060651f)
_J. Proteome Research_, *6* 1530-1539 (2007).

* E.H. Jeninga, O. van Beekum, A.D.J. van Dijk, N. Hamers, B.I. Hendriks-Stegeman, **A. M. J. J. Bonvin**, R. Berger and E. Kalkhoven
[Impaired Peroxisome Proliferator-Activated Receptor gamma Function through Mutation of a Conserved Salt Bridge (R425C) in Familial Partial Lipodystrophy.](https://doi.org/doi:10.1210/me.2006-0485)
_Molecular Endocrinology_, *21* 1049-1065 (2007).

* A. M. Wu, T. Singh, J.-H. Liu, M. Krzeminski, R. Russwurm, H.-C. Siebert, **A. M. J. J. Bonvin**, Sabine Andre, H.-J. Gabius
[Activity-structure correlations in divergent lectin-evolution: fine specificity of chicken galectin CG-14 and computational analysis of flexible ligand docking for CG-14 and the closely related CG-16.](https://doi.org/doi:10.1093/glycob/cwl062)
_Glycobiology_, *17* 165-184 (2007).


## 2006
<hr />

* A.D.J. van Dijk and **A.M.J.J. Bonvin**
[Solvated docking: introducing water into the modelling of biomolecular complexes.](https://doi.org/doi:10.1093/bioinformatics/btl395)
_Bioinformatics_, *22* 2340-2347 (2006).

* G.A. Jensen, O.M. Andersen, **A.M.J.J. Bonvin**, I. Bjerrum-Bohr, M. Etzerodt, H.C. Thogersen, F.M. Poulsen1 and B.B. Kragelund
[Binding Site Structure of One LRP/RAP Complex -Implications for a Common Ligand/Receptor Binding Motif.](https://doi.org/10.1016/j.jmb.2006.07.013)
_J. Mol. Biol._, *362* 700-716 (2006).

* S.J. de Vries and **A.M.J.J. Bonvin**
[Intramolecular surface contacts contain information about protein-protein interface regions.](https://doi.org/doi:10.1093/bioinformatics/btl275)
_Bioinformatics_, *22* 2094-2098 (2006).

* M. van Dijk, A.D.J. van Dijk, V. Hsu, R. Boelens and **A.M.J.J. Bonvin**
[Information-driven Protein-DNA Docking using HADDOCK: it is a matter of flexibility.](https://doi.org/doi:10.1093/nar/gkl412)
_Nucl. Acids Res._, *34* 3317-3325 (2006).

* J.P. Langedijk, G. Fuentes, R. Boshuizen and **A.M.J.J. Bonvin**
[Two-rung model of a left-handed beta-helix for prions explains species barrier and strain variation in TSEs.](https://doi.org/doi:10.1016/j.jmb.2006.05.042)
_J. Mol. Biol._, *360* 907-920 (2006).

* L. Rutten, J. Geurtsen, W. Lambert, J.J. Smolenaers, **A.M.J.J. Bonvin**, A. de Haan, P. van der Ley, M.R. Egmond, P. Gros and J. Tommassen.
[Crystal structure and catalytic mechanism of the LPS 3-O-deacylase PagL from Pseudomonas aeruginosa.](https://doi.org/doi:10.1073/pnas.0509392103)
_Proc. Natl. Acad. Sci._, *103* 7071-7076-244 (2006).

* N. Sibille, A. Favier, A.I. Azuaga, G. Ganshaw, R. Bott, **A.M.J.J. Bonvin**, R. Boelens &amp; N.A.J. van Nuland
[Comparative NMR study on the impact of point mutations on protein stability of Pseudomonas mendocina lipase.](https://doi.org/10.1110/ps.062213706)
_Prot. Sci._, *15* 1915-1927 (2006).

* E. AB, D.J.R. Pugh, R. Kaptein, R. Boelens and **A.M.J.J. Bonvin**
[Direct Use of Unassigned Resonances in NMR Structure Calculations with Proxy Residues](https://doi.org/10.1021/ja058504q)
_J. Am. Chem. Soc._, *128*, 7566-7571 (2006).

* A.D.J. van Dijk, R. Kaptein, R. Boelens and **A.M.J.J. Bonvin**
[Combining NMR relaxation with chemical shift perturbation data to drive protein-protein docking](https://doi.org/doi:10.1007/s10858-006-0024-8)
_J. Biomol. NMR_, *34*, 237-244 (2006).

* F.J. Fuchs, **A.M.J.J. Bonvin<sup>\*</sup>**, B. Bochicchio, A. Pepe, A.J.P. Alix and A.M. Tamburro
[Kinetics and thermodynamics of type VIII beta-turn formation: a CD, NMR and microsecond explicit molecular dynamics study of the GDNP tetrapeptide.](https://doi.org/doi:10.1529/biophysj.105.074401)
_Biophys. J._, *90*, 2745-2759 (2006).

* **A.M.J.J. Bonvin**
[Flexible protein-protein docking.](https://doi.org/doi:10.1016/j.sbi.2006.02.002)
_Curr. Opin. Struct. Biol._, *16*, 194-200 (2006).

* M.B. Kamphuis, **A.M.J.J. Bonvin**, M.C. Monti, M. Lemonnier, A. Munoz-Gomez, R.H.H. van den Heuvel, R. Diaz-Orejas and R. Boelens
[Model for RNA Binding and the Catalytic Site of the RNase Kid of the Bacterial parD Toxin-Antitoxin System.](https://doi.org/doi:10.1016/j.jmb.2005.12.033)
_J. Mol. Biol._, *357*, 115-126 (2006).

* S.J. de Vries, A.D.J. van Dijk and **A.M.J.J. Bonvin**
[WHISCY: WHat Information does Surface Conservation Yield? Application to data-driven docking.](https://doi.org/doi:10.1002/prot.20842)
_Proteins: Struc. Funct. &amp; Bioinformatics_, *63*, 479-489 (2006).

* S. Tomaselli, V. Esposito, P. Vangone, N.A.J. van Nuland, **A.M.J.J. Bonvin**, R. Guerrini, T. Tancredi, P.A. Temussi and D. Picone.
[The alpha-to-beta conformational transition of Alzheimer's Abeta-(1-42) peptide in aqueous media is reversible: a step by step conformational analysis suggests the location of beta conformation seeding.](https://doi.org/doi:10.1002/cbic.200500223)
_Chem. Bio. Chem._, *7*, 257-267 (2006).

* J.S. Grinstead, S.-T D. Hsu, W. Laan, **A.M.J.J. Bonvin**, K.J. Hellingwerf, R. Boelens and R. Kaptein
[The solution structure of the AppA BLUF domain: Insight into the mechanism of light-induced signaling.](https://doi.org/doi:10.1002/cbic.200500270)
_Chem. Bio. Chem_, *7*, 187-193 (2006).


## 2005
<hr />

* G. Fuentes, A.J. Nederveen, R. Kaptein, R. Boelens and **A.M.J.J. Bonvin**
[Describing partially unfolded states from sparse NMR data.](https://doi.org/doi:10.1007/s10858-005-3207-9)
_J. Biomol. NMR_, *33* 175-186 (2005).

* **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
[NMR analysis of protein interactions](https://doi.org/doi:10.1016/j.cbpa.2005.08.011)
_Curr. Opin. Chem. Biol._, *9*, 501-508 (2005).

* R.K. Salinas, G.E. Folkers, **A.M.J.J. Bonvin**, D. Das, R. Boelens and R. Kaptein.
[Altered specificity in DNA binding by the lac repressor: a mutant lac headpiece that minics the gal repressor.](https://doi.org/doi:10.1002/cbic.200500049)
_Chem. Bio. Chem._, *9*, 1628-1637 (2005).

* A.D.J. van Dijk, S.J. de Vries, C. Dominguez, H. Chen, H.-X. Zhou and **A.M.J.J. Bonvin**
[Data-driven docking: HADDOCKs adventures in CAPRI.](https://doi.org/doi:10.1002/prot.20563)
_Proteins: Struc. Funct. &amp; Bioinformatics_, *60*, 232-238 (2005).

* A.D.J. van Dijk, D. Fushman and **A.M.J.J. Bonvin**
[Various strategies of using residual dipolar couplings in NMR-driven protein docking: Application to Lys48-linked di-ubiquitin and validation against 15N-relaxation data.](https://doi.org/doi:10.1002/prot.20476)
_Proteins: Struc. Funct. &amp; Bioinformatics_, *60* 367-381(2005).

* A.J. Nederveen and **A.M.J.J. Bonvin**
[NMR relaxation and internal dynamics of ubiquitin from a 0.2 microsecond MD simulation.](https://doi.org/doi:10.1021/ct0498829)
_J. Chem. Theo. Comput._, *1*, 363-374 (2005).

* K.F. Lin, Y.N. Liu, S.-T. D. Hsu, D. Samuel, C.-H. Cheng, **A.M.J.J. Bonvin** and P.-C. Lyu
[Characterization and structural analyses of nsLTP1 from mung bean.](https://doi.org/doi:10.1021/bi047608v)
_Biochemistry_, *44*, 5703-5712 (2005).

* A.J. Nederveen, J.F. Doreleijers, W. Vranken, Z. Miller, C.A.E.M. Spronk, S.B. Nabuurs, P. Guntert, M. Livny, J.L. Markley, M. Nilges, E.L. Ulrich, R. Kaptein and **A.M.J.J. Bonvin**
[RECOORD: a REcalculated COORdinates Database of 500+ proteins from the PDB using restraints from the BioMagResBank](https://doi.org/10.1002/prot.20408)
_Proteins: Struc. Funct. &amp; Bioinformatics_, *59*, 662-672 (2005).

* J.F. Doreleijers, A.J. Nederveen, W. Vranken, J. Lin, **A.M.J.J. Bonvin,** R. Kaptein, J.L. Markley and E.L. Ulrich
[BioMagResBank databases DOCR and FRED with converted and filtered sets of experimental NMR restraints and coordinates from over 500 protein PDB structures.](https://doi.org/10.1007/s10858-005-2195-0)
_J. Biomol. NMR_, *32*, 1-12 (2005).

* A.N. Volkov, D. Ferrari, J.A.R. Worrall, **A.M.J.J. Bonvin** and M. Ubbink
[The orientations of cytochrome c in the highly dynamic complex with cytochrome b5 visualized by NMR and docking using HADDOCK.](https://doi.org/doi:10.1110/ps.041150205)
_Protein Science_, *14*, 799-811 (2005).

* A.D.J. van Dijk, R. Boelens and **A.M.J.J. Bonvin**
[Data-driven docking for the study of biomolecular complexes.](https://doi.org/doi:10.1111/j.1742-4658.2004.04473.x)
_FEBS Journal_, *272*, 293-312 (2005).

* S.-T. D. Hsu, C. Peter, W.F. van Gunsteren and **A.M.J.J. Bonvin**
[Entropy calculation of HIV-1 Env gp120, its receptor CD4 and their complex: an analysis of entropy changes upon complexation.](https://doi.org/doi:10.1529/biophysj.104.044933)
_Biophys. J._, *88*, 15-24 (2005).


## 2004
<hr />

* K. Houben, C. Dominguez, F.M.A. van Schaik, H.Th.M. Timmers, **A.M.J.J. Bonvin** and R. Boelens
[Solution structure of the ubiquitin conjugating enzyme UbcH5B.](https://doi.org/doi:10.1016/j.jmb.2004.09.054)
_J. Mol. Biol._, *344*, 513-526 (2004).

* S.-T. D. Hsu, E. Breukink, E. Tischenko, M.A.G. Lutters, B. de Kruijff, R. Kaptein, **A.M.J.J. Bonvin** and N.A.J. van Nuland
[The nisin-lipid II complex reveals a pyrophosphate cage that provides a blueprint for novel antibiotics](https://doi.org/doi:10.1038/nsmb830)
_Nature Struc. &amp; Mol. Biol._, *11*, 963-967 (2004).

* A.G. Tzakos, P. Fuchs, N.A.J. van Nuland, A. Troganis, T. Tselios, S. Deraos, J. Matsoukas, I.P. Gerothanassis and **A.M.J.J. Bonvin**
[NMR and MD Studies of an Autoimmune MBP Peptide and its Antagonist. Structural Implications for the MHC II (I-Au) Peptide Complex through Docking Calculations.](https://doi.org/doi:10.1111/j.1432-1033.2004.04274.x)
_Eur. J. Biochem._,*271*, 3399-3413 (2004).

* C.G. Kalodimos, N. Biris,**A.M.J.J. Bonvin**, M.M. Levandoski, M. Guennuegues, R. Boelens, R. Kaptein
[Structure and flexibility adaptation between nonspecific and specific protein-DNA complexes.](https://www.sciencemag.org/cgi/content/abstract/305/5682/386)
_Science_, *205* 386-389 (2004).

* S.B. Nabuurs, A.J. Nederveen, W. Wranken, J.F. Doreleijers, **A.M.J.J. Bonvin**, G.W.Vuister, G. Vriend and C.A.E.M. Spronk
[DRESS, a Database of REfined Solution nmr Structures.](https://doi.org/10.1002/prot.20118)
_Proteins: Struc. Funct. &amp; Bioinformatics_, *55* 483-486 (2004).

* S.-T.D. Hsu and **A.M.J.J. Bonvin**
[Atomic insight into the CD4 binding-induced conformational changes in HIV-1 gp120.](https://doi.org/10.1002/prot.20061)
_Proteins: Struc. Funct. &amp; Bioinformatics_, *55* 582-593 (2004).

* C. Dominguez, **A.M.J.J. Bonvin**, G.S. Winkler, F.M.A. van Schaik, H.Th.M. Timmers and R.Boelens
[Structural model of the UbcH5B/CNOT4 complex revealed by combining NMR, mutagenesis and docking approaches.](https://doi.org/10.1016/j.str.2004.03.004)
_Structure_, *12* 633-644 (2004).

* F. Arnesano, L. Banci, I. Bertini and **A.M.J.J. Bonvin**
[A docking approach to the study of copper trafficking proteins: interaction between metallochaperones and soluble domains of copper ATPases.](https://doi.org/10.1016/j.str.2004.03.006)
_Structure_, *12* 669-676 (2004).

* A. van Drogen-Petit, C. Zwahlen, M. Peter and **A.M.J.J. Bonvin**
[Insight into molecular interactions between two PB1 domains.](https://doi.org/doi:10.1016/j.jmb.2003.12.062)
_J. Mol. Biol._, *336/5* 1195-1210 (2004).


## 2003
<hr />

* P. Fuchs, **A.M.J.J. Bonvin**, A. Pepe, A.J.P. Alix and A.M. Tamburro
Spectroscopic and theoretical characterisation of the type VIII beta-turn. Study of the tetrapeptide GDNP.
In: Elastin 2002, edited by A.M. Tamburro and A. Pepe. EditricErmes, Potenza, Italy, pp. 81-88 (2003).

* L.M.I. Koharudin, **A.M.J.J. Bonvin**, R. Kaptein and R. Boelens
[Use of very long range NOEs in a fully deuterated protein: an approach for rapid protein fold determination.](https://doi.org/10.1016/S1090-7807(03)00149-6)
_J. Magn. Reson._, *163*, 228-235 (2003).

* C.J. Oomen, P. Hoogerhout, **A.M.J.J. Bonvin**, B. Kuipers, H. Brugghe, H. Timmermans, S.R. Haseley, L. van Alphen and P. Gros
[Immunogenicity of peptide-vaccine candidates predicted by molecular dynamics simulations.](https://doi.org/10.1016/S0022-2836(03)00377-2)
_J. Mol. Biol._, *328*, 1083-1089 (2003).

* S.-T. D. Hsu, E. Breukink, G. Bierbaum, H.-G. Sahl, D. de Kruijff, R. Kaptein, N.A.J. van Nuland and **A.M.J.J. Bonvin**
[NMR study of mersacidin and lipid II interaction in dodecylphosphocholine micelles. Conformational changes are a key to antimicrobial activity.](https://www.jbc.org/cgi/content/abstract/278/15/13110)
_J. Biol. Chem._, *278*, 13110-13117 (2003).

* T. Stockner, H. Sterk, R. Kaptein and **A.M.J.J. Bonvin**
[Molecular dynamics simulations of a molecular switch in the glucocorticoid receptor.](https://doi.org/10.1016/S0022-2836(03)00316-4)
_J. Mol. Biol._, *328*, 325-334 (2003).

* A.G. Tzakos, **A.M.J.J. Bonvin**, A. Troganis, P. Cordopatis, M.L. Amzel, I.P. Gerothanassis and N.A.J. van Nuland
[On the molecular basis of the recognition of angiotensin II (AII): NMR structure of AII in solution versus the X-ray structure of AII bound to the monoclonal antibody Fab131.](https://www.ejbiochem.org/cgi/content/abstract/270/5/849)
_Eur. J. Biochem._, *270*, 849-860 (2003).

* C. Dominguez, R. Boelens and **A.M.J.J. Bonvin**
[HADDOCK: A protein-protein docking approach based on biochemical or biophysical information.](https://pubs.acs.org/doi/abs/10.1021/ja026939x)
_J. Am. Chem. Soc._, *125*, 1731-1737 (2003).

* J. Linge, M.A. Williams, C.A.E.M. Spronk, **A.M.J.J. Bonvin**<sup>\*</sup> and M. Nilges<sup>\*</sup>
[Refinement of protein structures in explicit solvent.](https://www3.interscience.wiley.com/cgi-bin/abstract/102523901/START)
_PROTEINS: Struct. Funct. &amp; Genetics_ *50*, 496-506 (2003).

* C.A.E.M. Spronk, S.B. Nabuurs, **A.M.J.J. Bonvin**, E. Krieger, G.W. Vuister and G. Vriend
[The precision of NMR structure ensembles revisited.](https://doi.org/10.1023/A:1022819716110)
_J. Biomol. NMR_ *25*, 225-234 (2003).


## 2002
<hr />

* C.G. Kalodimos, **A.M.J.J. Bonvin**, R.K. Salinas, R. Wechselberger, R. Boelens and R. Kaptein
[Plasticity in protein-DNA recognition: lac repressor interacts with its natural operator O1 through alternative conformations of its DNA-binding domain.](https://emboj.oupjournals.org/cgi/content/abstract/21/12/2866)
_EMBO J._ *21*, 2866-2876 (2002).

* S.-T. Hsu, E. Breukink, B. de Kruiff, R. Kaptein, **A.M.J.J. Bonvin** and N. van Nuland
[Mapping the targeted membrane pore formation mechanism by solution NMR: the nisin Z and lipid II interaction in SDS micelles.](https://pubs.acs.org:80/journals/bichaw/abstract.cgi/bichaw/2002/41/i24/abs/bi025679t.html)
_Biochemistry_ *41*, 7670-7676 (2002).

* S. Singh, G.E. Folkers, **A.M.J.J. Bonvin**, R. Boelens, R. Wechselberger, A. Niztayev and R. Kaptein
[Solution structure and DNA binding properties fo the C-terminal domain of UvrC from <i>E-coli</i>.](https://emboj.oupjournals.org/cgi/content/abstract/21/22/6257)
_EMBO J._, *21*, 6257-6266 (2002).

* N. D'Amelio, **A.M.J.J Bonvin**, M. Czisch, P. Barker and R. Kaptein
[The C terminus of apocytocrome b562 undergoes fast motions and slow exchange among ordered conformations resembling the folded state](https://pubs.acs.org:80/journals/bichaw/abstract.cgi/bichaw/2002/41/i17/abs/bi011863n.html)
_Biochemistry_, *41*, 5505-5514 (2002).


## 2001
<hr />

* **A.M.J.J. Bonvin**, K. Houben, M. Guenneugues, R. Kaptein and R. Boelens
[Rapid protein fold determination using secondary chemical shifts and cross-hydrogen bond 15N-13C' scalar couplings (<sup>3hb</sup>J<sub>NC'</sub>).](https://www.ncbi.nlm.nih.gov:80/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;list_uids=11775739&amp;dopt=Abstract)
_J. Biomol. NMR_, *21*, 221-233 (2001).


## 2000
<hr />

* M. van Beest, D. Dooijes, M. van Wetering, S. Kjaerulff, **A.M.J.J. Bonvin**, O. Nielsen and H. Clevers
[Sequence-specific HMG box factors recognize 10-12 base pair minor groove motifs.](https://www.jbc.org/cgi/content/abstract/M004102200v1)
_J. Biol. Chem._ *275*, 27266-27273 (2000).

* G. Melacini, **A.M.J.J. Bonvin**, M. Goodman, R. Boelens and R. Kaptein
[Hydration dynamics of the collagen triple-helix by NMR.](https://www.idealibrary.com/links/doi/10.1006/jmbi.2000.3919)
_J. Mol. Biol._ *300*, 1041-1049 (2000).

* P.J.A. van Tilborg, M. Czisch, F.A.A. Mulder, G.E. Folkers, **A.M.J.J. Bonvin**, M. Nair, R. Boelens and R. Kaptein
[Changes in dynamical behaviour of the retinoid X receptor DNA-binding domain upon binding to 14 base pair DNA half site.](https://pubs.acs.org/CHECKCCIP-1016438394/isubscribe/journals/alljournals/jtext.cgi?bichaw/39/i30/abs/bi991550g)
_Biochemistry_ *39*, 8747-8757 (2000).

* **A.M.J.J. Bonvin**, A.E. Mark and W.F. van Gunsteren
[The GROMOS96 benchmarks for molecular simulation.](https://www.elsevier.nl/gej-ng/10/15/40/58/27/26/abstract.html)
_Comput. Phys. Commun._ *128*, 550-557 (2000).

* **A.M.J.J. Bonvin**
[Localisation and dynamics of sodium counterions around DNA in solution from molecular dynamics simulations.](https://link.springer.de/link/service/journals/00249/bibs/0029001/00290057.htm)
_Eur. Biophys. J._ *29*, 57-60 (2000).  (<a href="https://link.springer.de/link/service/journals/00249/supp/2000/061/061.htm">Supplementary material</a>)

* **A.M.J.J. Bonvin** and W.F. van Gunsteren
[b-hairpin stability and folding: Molecular dynamics studies of the first b-hairpin of tendamistat.](https://doi.org/doi:10.1006/jmbi.1999.3446)
_J. Mol. Biol._ *296* 255-268 (2000).

* C.J. Oomen, **A.M.J.J. Bonvin**, S.R. Haseley, P. Hoogerhout, L. van Alphen, J. Kroon and P. Gros
Rational design of a subtype-specific peptide vaccine against Neisseria meningitis"
In: Proceedings of the 16th American Peptide Symposium: Peptides for a new millenium.  ESCOM, Leiden, the Netherlands, 2000, pp. 702-703.


## 1999
<hr />

* C.A.E.M. Spronk, **A.M.J.J. Bonvin**, P.K. Radha, G. Melacini, R. Boelens and R. Kaptein
[The solution structure of lac repressor headpiece 62 complexed to a symmetrical lac operator sequence determined by NMR and restrained molecular dynamics.](https://doi.org/doi:10.1016/S0969-2126(00)88339-2)
_Structure_ *7* 6472-6480 (1999).

* W.F. van Gunsteren, **A.M.J.J. Bonvin**, X. Daura, L.J. Smith
Aspects of modelling biomolecular structure on the basis of spectroscopic or diffraction data.
In: Modern techniques in protein NMR, Krishna and Berliner eds.,  Plenum Publ. Corp. (1999), New York


## 1998
<hr />

* **A.M.J.J. Bonvin**, M. Sunnerhagen, G. Otting and W.F. van Gunsteren
[Water molecules in DNA recognition II: A molecular dynamics view of the structure and hydration of the trp operator.](https://doi.org/doi:10.1006/jmbi.1998.2034)
_J. Mol. Biol._ *282*, 859-873 (1998).

* M. Sunnerhagen, V.P. Denisov, K. Venu, **A.M.J.J. Bonvin**, J. Carey, B. Halle and G. Otting
[Water molecules in DNA recognition I: Hydration lifetimes of the trp operator in solution measured by NMR spectroscopy.](https://doi.org/doi:10.1006/jmbi.1998.2033)
_J. Mol. Biol._ *282*, 847-858 (1998).


## 1997
<hr />

* R. Kaptein, **A.M.J.J. Bonvin** and R. Boelens
Biomolecular structure and dynamics: recent experimental and theoretical advances.
In: Proceedings of the Nato Summer School. Eds G. Vergoten and T. Theophanides  Kluwer Academic Publisher, 189-209 (1997).


## 1996
<hr />

* **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
Protein Structures: Relaxation matrix refinement.
In: Encyclopedia of Nuclear Magnetic Resonance Vol. 6. Eds D.M. Grant and R.K. Harris,  John Wiley &amp; Sons, 3801-3811 (1996).

* M. Slijper, **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
[Refined structure of lac repressor headpiece (1-56) determined by relaxation matrix calculations from 2D and 3D NOE data: change of tertiary structure upon binding to the lac operator.](https://doi.org/doi:10.1006/jmrb.1995.1094)
_J. Mol. Biol._ *259*, 761-773 (1996).

* **A.M.J.J. Bonvin** and A.T. Brunger
[Do NOE distances contain enough information to assess the relative populations of multi-conformer structures?](https://doi.org/doi:10.1007/BF00190458)
_J. Biomol. NMR._ *7*, 72-76 (1996).


## 1995
<hr />

* M. Slijper, **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
[Application of structure refinement using 3D NOE-NOE spectroscopy to lac repressor headpiece (1-56).](https://doi.org/doi:10.1006/jmrb.1995.1094)
_J. Magn. Reson._ *B 107*, 298-301 (1995).

* **A.M.J.J. Bonvin** and A.T. Brunger
[Conformational variability of solution nuclear magnetic resonance structures.](https://doi.org/doi:10.1006/jmbi.1995.0360)
_J. Mol. Biol._ *250*, 80-93 (1995).

* M.A.A. van Tilborg, **A.M.J.J. Bonvin**, K. Hard, A.L. Davis, B. Maler, R. Boelens and R. Kaptein
[Structure refinement of the glucocorticoid receptor DNA-binding domain from NMR data by relaxation matrix calculations.](https://doi.org/doi:10.1006/jmbi.1995.0173)
_J. Mol. Biol._ *247*, 689-700 (1995).


## 1994
<hr />

* **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
[Time- and ensemble-averaged direct NOE restraints.](https://doi.org/doi:10.1007/BF00178343)
_J. Biomol. NMR_ *4*, 143-149 (1994).

* **A.M.J.J. Bonvin**, H. Vis, J.N. Breg, M.J.M. Burgering, R. Boelens and R. Kaptein
[NMR solution structure of the Arc repressor using relaxation matrix calculations.](https://doi.org/doi:10.1006/jmbi.1994.1138)
_J. Mol. Biol._ *236*, 328-341 (1994).

* **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
[Direct nuclear Overhauser effect refinement of crambin from 2D NMR data using a slow-cooling simulated annealing.](https://doi.org/doi:10.1002/bip.360340106)
_Biopolymers_ *34*, 39-50 (1994).


## 1993
<hr />

* **A.M.J.J. Bonvin**
Determination of biomolecular structures by NMR. Use of relaxation matrix calculations.
PhD Thesis, Utrecht University (1993) ISBN 90-393-0408-4.

* R.M.A. Knegtel, M. Katahira, J.G. Schilthuis, **A.M.J.J. Bonvin**, R. Boelens, D. Eib, CHECKCCIP.T. der Saag and R. Kaptein
NMR studies of the human retinoic acid receptor-b DNA-binding domain: metal coordination and three dimensional structure.
In: "Zinc-finger proteins in oncogenesis. DNA-binding and gene regulation".  M. Sluyser, G. AB, A.O. Brinkmann and R.A. Blankenstein (eds),  _Annals of the New York Academy of Sciences_ *684*, New York, 49-62 (1993).

* **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
Determination of biomolecular structures by NMR. Use of relaxation matrix calculations.
In:"computer simulations of biomolecular systems: theoretical and  experimental applications". Vol.2. W.F. van Gunsteren, P.K. Weiner and  A.J. Wilkinson (eds), ESCOM Science Publishers, 407-440 (1993).

* **A.M.J.J. Bonvin**, J.A.C. Rullmann, R.M.J.N. Lamerichs, R. Boelens and R. Kaptein
[Ensemble Iterative Relaxation Matrix Approach: a new NMR refinement protocol applied to the solution structure of crambin.](https://doi.org/doi:10.1002/prot.340150406)
_PROTEINS: Structure, Function &amp; Genetics_ *15*, 385-400 (1993).

* R.M.A. Knegtel, M. Katahira, J.G. Schilthuis, **A.M.J.J. Bonvin**, R. Boelens, D. Eib, P.T. der Saag and R. Kaptein
[The solution structure of the human retinoic acid receptor-b DNA-binding domain.](https://doi.org/doi:10.1007/BF00242472)
_J. Biomol. NMR_ *3*, 1-17 (1993).


## 1992
<hr />

* J.A.C. Rullmann, **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
Structure determination from NMR - Application to crambin
In: Computation of biomolecular structures: achievements, problems and perspectives."  (Proceeding of a workshop held at Schloss Ringberg, March 1990), T.M. Jovin and  D.M. Soumpasis (eds.), Springer, 1-14 (1992).


## 1991
<hr />

* **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
[Direct structure refinement using 3D NOE-NOE spectra of biomolecules.](https://doi.org/doi:10.1016/0022-2364(91)90180-2)
_J. Magn. Reson._ *95*, 626-631 (1991).

* **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
[Direct NOE refinement of biomolecular structures using 2D NMR data.](https://doi.org/doi:10.1007/BF01875523)
_J. Biomol. NMR_ *1*, 305-309 (1991).

* C. Gonzales, J.A.C. Rullmann, **A.M.J.J. Bonvin**, R. Boelens and R. Kaptein
[Toward an NMR R factor.](https://doi.org/doi:10.1016/0022-2364(91)90397-C)
_J. Magn. Reson._ *91*, 659-664 (1991).


## 1990
<hr />

* I. Burghardt , L. Di Bari, **A. Bonvin** and G. Bodenhausen
[Effect of strong coupling in multiple-quantum-filtered two-dimensional NOE spectroscopy.](https://doi.org/doi:10.1016/0022-2364(90)90044-A)
_J. Magn. Reson._ *86*, 652-656 (1990).

